
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A Snakemake-based bioinformatics toolkit for analyzing sequencing data">
      
      
        <meta name="author" content="Milne Group">
      
      
        <link rel="canonical" href="https://milne-group.github.io/SeqNado/configuration/">
      
      
        <link rel="prev" href="../genomes/">
      
      
        <link rel="next" href="../design/">
      
      
        
      
      
      <link rel="icon" href="https://raw.githubusercontent.com/Milne-Group/SeqNado/main/containers/pipeline/seqnado.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.2">
    
    
      
        <title>Configuration - SeqNado</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.484c7ddc.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../assets/_mkdocstrings.css">
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="green" data-md-color-accent="light-green">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#configuration" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="SeqNado" class="md-header__button md-logo" aria-label="SeqNado" data-md-component="logo">
      
  <img src="https://raw.githubusercontent.com/Milne-Group/SeqNado/main/containers/pipeline/seqnado.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            SeqNado
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Configuration
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="green" data-md-color-accent="light-green"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5s-1.65.15-2.39.42zM3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29zm.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14zM20.65 7l-1.77 3.79a7.02 7.02 0 0 0-2.38-4.15zm-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29zM12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="teal" data-md-color-accent="lime"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/Milne-Group/SeqNado" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    Milne-Group/SeqNado
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href=".." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../quick_start/" class="md-tabs__link">
          
  
  
    
  
  Getting Started

        </a>
      </li>
    
  

      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="./" class="md-tabs__link">
          
  
  
    
  
  Usage

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../api/" class="md-tabs__link">
          
  
  
    
  
  Reference

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../troubleshooting/" class="md-tabs__link">
          
  
  
    
  
  Help

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="SeqNado" class="md-nav__button md-logo" aria-label="SeqNado" data-md-component="logo">
      
  <img src="https://raw.githubusercontent.com/Milne-Group/SeqNado/main/containers/pipeline/seqnado.png" alt="logo">

    </a>
    SeqNado
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/Milne-Group/SeqNado" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    Milne-Group/SeqNado
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Getting Started
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Getting Started
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../quick_start/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Quick Start
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../installation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Installation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../initialisation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Initialisation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../genomes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Genome Setup
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" checked>
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Usage
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Usage
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Configuration
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Configuration
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#what-configuration-does" class="md-nav__link">
    <span class="md-ellipsis">
      
        What Configuration Does
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#information-to-know-before-configuration" class="md-nav__link">
    <span class="md-ellipsis">
      
        Information to Know Before Configuration
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#questions-asked-during-configuration" class="md-nav__link">
    <span class="md-ellipsis">
      
        Questions Asked During Configuration
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Questions Asked During Configuration">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#global-questions-all-assays-except-snp-crispr" class="md-nav__link">
    <span class="md-ellipsis">
      
        Global Questions (All Assays Except SNP &amp; CRISPR)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#atac-seq-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        ATAC-seq Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#chip-seq-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        ChIP-seq Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#cuttag-cat-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        CUT&amp;Tag (CAT) Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#rna-seq-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        RNA-seq Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mcc-capture-c-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        MCC (Capture-C) Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#methylation-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        Methylation Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#snp-calling-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        SNP Calling Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#crispr-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        CRISPR Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#understanding-multi-select-questions" class="md-nav__link">
    <span class="md-ellipsis">
      
        Understanding Multi-Select Questions
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#resource-constraints-to-consider" class="md-nav__link">
    <span class="md-ellipsis">
      
        Resource Constraints to Consider
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-seqnado-config" class="md-nav__link">
    <span class="md-ellipsis">
      
        Running seqnado config
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Running seqnado config">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assay-types" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assay Types
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#example-usage" class="md-nav__link">
    <span class="md-ellipsis">
      
        Example Usage
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Example Usage">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#build-a-configuration-for-chip-seq" class="md-nav__link">
    <span class="md-ellipsis">
      
        Build a Configuration for ChIP-seq
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#interactive-multiomics-configuration-multiple-assays" class="md-nav__link">
    <span class="md-ellipsis">
      
        Interactive Multiomics Configuration (Multiple Assays)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#third-party-tools" class="md-nav__link">
    <span class="md-ellipsis">
      
        Third Party Tools
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Third Party Tools">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#supported-tools-reference" class="md-nav__link">
    <span class="md-ellipsis">
      
        Supported Tools Reference
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Supported Tools Reference">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#alignment-indexing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Alignment &amp; Indexing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#read-processing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Read Processing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#peak-calling-chip-seq-atac-seq-cuttag-mcc" class="md-nav__link">
    <span class="md-ellipsis">
      
        Peak Calling (ChIP-seq, ATAC-seq, CUT&amp;Tag, MCC)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quantification-rna-seq" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quantification (RNA-seq)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bigwig-generation-visualization" class="md-nav__link">
    <span class="md-ellipsis">
      
        Bigwig Generation &amp; Visualization
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#motif-analysis-peak-regions" class="md-nav__link">
    <span class="md-ellipsis">
      
        Motif Analysis (peak regions)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variant-calling-snp" class="md-nav__link">
    <span class="md-ellipsis">
      
        Variant Calling (SNP)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#methylation-specialized" class="md-nav__link">
    <span class="md-ellipsis">
      
        Methylation &amp; Specialized
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#data-analysis-normalisation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Data Analysis &amp; Normalisation
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#configuration-logic-tool-interactions" class="md-nav__link">
    <span class="md-ellipsis">
      
        Configuration Logic &amp; Tool Interactions
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#common-pitfalls-and-best-practices" class="md-nav__link">
    <span class="md-ellipsis">
      
        Common Pitfalls and Best Practices
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Common Pitfalls and Best Practices">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#pitfalls-to-avoid" class="md-nav__link">
    <span class="md-ellipsis">
      
        Pitfalls to Avoid
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#best-practices" class="md-nav__link">
    <span class="md-ellipsis">
      
        Best Practices
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../design/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Design Guide
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../geo_download/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GEO/SRA Download
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pipeline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Pipeline Overview
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../outputs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Outputs
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Output Examples
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../normalisation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Normalisation Methods
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tools/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Third-Party Tools
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../cli/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    CLI Reference
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_4" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../api/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Reference
  

    
  </span>
  
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_4" id="__nav_4_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Reference
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/core/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Core Types
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/config/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Configuration API
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/inputs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Inputs API
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/outputs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Outputs API
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../assays/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Assay Configurations
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tool-config-examples/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Tool Config Examples
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../citation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Citation Guidelines
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Help
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Help
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../troubleshooting/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Troubleshooting
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../cluster_config/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    HPC Clusters
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../faq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    FAQ
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#what-configuration-does" class="md-nav__link">
    <span class="md-ellipsis">
      
        What Configuration Does
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#information-to-know-before-configuration" class="md-nav__link">
    <span class="md-ellipsis">
      
        Information to Know Before Configuration
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#questions-asked-during-configuration" class="md-nav__link">
    <span class="md-ellipsis">
      
        Questions Asked During Configuration
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Questions Asked During Configuration">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#global-questions-all-assays-except-snp-crispr" class="md-nav__link">
    <span class="md-ellipsis">
      
        Global Questions (All Assays Except SNP &amp; CRISPR)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#atac-seq-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        ATAC-seq Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#chip-seq-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        ChIP-seq Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#cuttag-cat-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        CUT&amp;Tag (CAT) Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#rna-seq-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        RNA-seq Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mcc-capture-c-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        MCC (Capture-C) Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#methylation-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        Methylation Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#snp-calling-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        SNP Calling Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#crispr-specific" class="md-nav__link">
    <span class="md-ellipsis">
      
        CRISPR Specific
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#understanding-multi-select-questions" class="md-nav__link">
    <span class="md-ellipsis">
      
        Understanding Multi-Select Questions
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#resource-constraints-to-consider" class="md-nav__link">
    <span class="md-ellipsis">
      
        Resource Constraints to Consider
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-seqnado-config" class="md-nav__link">
    <span class="md-ellipsis">
      
        Running seqnado config
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Running seqnado config">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assay-types" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assay Types
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#example-usage" class="md-nav__link">
    <span class="md-ellipsis">
      
        Example Usage
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Example Usage">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#build-a-configuration-for-chip-seq" class="md-nav__link">
    <span class="md-ellipsis">
      
        Build a Configuration for ChIP-seq
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#interactive-multiomics-configuration-multiple-assays" class="md-nav__link">
    <span class="md-ellipsis">
      
        Interactive Multiomics Configuration (Multiple Assays)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#third-party-tools" class="md-nav__link">
    <span class="md-ellipsis">
      
        Third Party Tools
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Third Party Tools">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#supported-tools-reference" class="md-nav__link">
    <span class="md-ellipsis">
      
        Supported Tools Reference
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Supported Tools Reference">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#alignment-indexing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Alignment &amp; Indexing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#read-processing" class="md-nav__link">
    <span class="md-ellipsis">
      
        Read Processing
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#peak-calling-chip-seq-atac-seq-cuttag-mcc" class="md-nav__link">
    <span class="md-ellipsis">
      
        Peak Calling (ChIP-seq, ATAC-seq, CUT&amp;Tag, MCC)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quantification-rna-seq" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quantification (RNA-seq)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bigwig-generation-visualization" class="md-nav__link">
    <span class="md-ellipsis">
      
        Bigwig Generation &amp; Visualization
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#motif-analysis-peak-regions" class="md-nav__link">
    <span class="md-ellipsis">
      
        Motif Analysis (peak regions)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variant-calling-snp" class="md-nav__link">
    <span class="md-ellipsis">
      
        Variant Calling (SNP)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#methylation-specialized" class="md-nav__link">
    <span class="md-ellipsis">
      
        Methylation &amp; Specialized
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#data-analysis-normalisation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Data Analysis &amp; Normalisation
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#configuration-logic-tool-interactions" class="md-nav__link">
    <span class="md-ellipsis">
      
        Configuration Logic &amp; Tool Interactions
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#common-pitfalls-and-best-practices" class="md-nav__link">
    <span class="md-ellipsis">
      
        Common Pitfalls and Best Practices
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Common Pitfalls and Best Practices">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#pitfalls-to-avoid" class="md-nav__link">
    <span class="md-ellipsis">
      
        Pitfalls to Avoid
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#best-practices" class="md-nav__link">
    <span class="md-ellipsis">
      
        Best Practices
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
                



  


  <nav class="md-path" aria-label="Navigation" >
    <ol class="md-path__list">
      
        
  
  
    <li class="md-path__item">
      <a href=".." class="md-path__link">
        
  <span class="md-ellipsis">
    Home
  </span>

      </a>
    </li>
  

      
      
        
  
  
    
    
      <li class="md-path__item">
        <a href="./" class="md-path__link">
          
  <span class="md-ellipsis">
    Usage
  </span>

        </a>
      </li>
    
  

      
    </ol>
  </nav>

              
              <article class="md-content__inner md-typeset">
                
                  


  
    <a href="https://github.com/Milne-Group/SeqNado/edit/main/docs/configuration.md" title="Edit this page" class="md-content__button md-icon" rel="edit">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75z"/></svg>
    </a>
  
  
    
      
    
    <a href="https://github.com/Milne-Group/SeqNado/raw/main/docs/configuration.md" title="View source of this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0 8a5 5 0 0 1-5-5 5 5 0 0 1 5-5 5 5 0 0 1 5 5 5 5 0 0 1-5 5m0-12.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5"/></svg>
    </a>
  


<p><a href="../">← Back to main page</a></p>
<h1 id="configuration">Configuration<a class="headerlink" href="#configuration" title="Anchor link to this section">&para;</a></h1>
<p>Build your analysis configuration after genome setup (see <a href="../genomes/">Genomes</a> for genome configuration).</p>
<h2 id="what-configuration-does">What Configuration Does<a class="headerlink" href="#what-configuration-does" title="Anchor link to this section">&para;</a></h2>
<p>The <code>seqnado config</code> command generates a YAML configuration file that defines all workflow parameters for your analysis. This file acts as a blueprint for the entire pipeline run:</p>
<ul>
<li><strong>Specifies tools and versions</strong> to use for each step (alignment, peak calling, quantification, etc.)</li>
<li><strong>Sets tool-specific parameters</strong> (e.g., peak caller mode, strandedness, normalisation method)</li>
<li><strong>Defines output locations</strong> and naming conventions</li>
<li><strong>Captures assay-specific choices</strong> (which normalisation method? which peak caller? which pileup method?)</li>
<li><strong>Enables reproducibility</strong> by version-locking the configuration and making it version-controlled</li>
</ul>
<p>The configuration file is generated based on your assay type. You can review, edit, and customize it before running the pipeline. Changes to the config do <strong>not</strong> affect samples already processed — each analysis can use different parameters by pointing to different config files.</p>
<h2 id="information-to-know-before-configuration">Information to Know Before Configuration<a class="headerlink" href="#information-to-know-before-configuration" title="Anchor link to this section">&para;</a></h2>
<p><strong>⚠️ CRITICAL: Before running <code>seqnado config</code>, you must first run <code>seqnado init</code> to configure your reference genome(s). See <a href="../genomes/">Genomes</a>.</strong></p>
<p>Once you have configured genomes, you'll be asked about your configuration choices during <code>seqnado config</code>. SeqNado automatically detects some parameters from your FASTQ files (sequencing type, read length, adapter content), but you should know these details upfront:</p>
<table>
<thead>
<tr>
<th>Information</th>
<th>Why It Matters</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Reference genome &amp; build</strong></td>
<td>REQUIRED — Configure via <code>seqnado init</code> before running config</td>
<td>hg38, mm39, mm10, dm6, or custom build</td>
</tr>
<tr>
<td><strong>Strandedness</strong> (RNA-seq only)</td>
<td>SeqNado will ask during config; check with sequencing facility</td>
<td>Unstranded (0), forward (1), or reverse (2) — usually dUTP-based kits are reverse-stranded</td>
</tr>
<tr>
<td><strong>Spike-in species</strong> (if applicable)</td>
<td>You'll specify this during config; confirm what you used</td>
<td><em>Drosophila</em> dm6, lambda phage, methylation controls (Lambda, 250bp-v1, 2kb-unmod), etc.</td>
</tr>
<tr>
<td><strong>Control/input samples</strong> (ChIP-seq only)</td>
<td>Not asked during config, but required in <a href="../design/">Design</a> file if using with-input normalisation</td>
<td>Do you have paired input controls? Name them as <code>samplename_input</code> in design file</td>
</tr>
<tr>
<td><strong>GTF annotation</strong> (RNA-seq only)</td>
<td>You'll need this for quantification; verify it matches your genome build</td>
<td>GRCh38 GTF for hg38; mm10 GTF for mm10</td>
</tr>
<tr>
<td><strong>Peak type expectations</strong> (ChIP/ATAC/CAT)</td>
<td>Helps you choose peak caller during config</td>
<td>Narrow (TFs, H3K4me3) vs broad (H3K27me3, H3K36me3)</td>
</tr>
</tbody>
</table>
<hr />
<h2 id="questions-asked-during-configuration">Questions Asked During Configuration<a class="headerlink" href="#questions-asked-during-configuration" title="Anchor link to this section">&para;</a></h2>
<p>The <code>seqnado config</code> command guides you through a series of interactive questions tailored to your assay type. Below are the complete questions for each assay, with defaults shown in parentheses.</p>
<h3 id="global-questions-all-assays-except-snp-crispr">Global Questions (All Assays Except SNP &amp; CRISPR)<a class="headerlink" href="#global-questions-all-assays-except-snp-crispr" title="Anchor link to this section">&para;</a></h3>
<p>These questions appear for ATAC, ChIP, CAT, RNA, MCC, and Methylation:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>Make Bigwigs? (default: yes)
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>  ├─ Bigwig method(s) — comma-separated (default: deeptools; options: deeptools, bamnado)
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>  ├─ Binsize for bigwigs (default: 10)
</span><span id="__span-0-4"><a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>  └─ Bigwig scaling method(s) — comma-separated (default: unscaled; options: unscaled, csaw)
</span><span id="__span-0-5"><a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>
</span><span id="__span-0-6"><a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a>Perform plotting? (default: no)
</span><span id="__span-0-7"><a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a>  ├─ Path to coordinates BED file (optional)
</span><span id="__span-0-8"><a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a>  └─ Path to genes BED file (optional)
</span><span id="__span-0-9"><a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a>
</span><span id="__span-0-10"><a id="__codelineno-0-10" name="__codelineno-0-10" href="#__codelineno-0-10"></a>Make heatmaps? (default: no)
</span><span id="__span-0-11"><a id="__codelineno-0-11" name="__codelineno-0-11" href="#__codelineno-0-11"></a>
</span><span id="__span-0-12"><a id="__codelineno-0-12" name="__codelineno-0-12" href="#__codelineno-0-12"></a>Generate GEO submission files? (default: no)
</span><span id="__span-0-13"><a id="__codelineno-0-13" name="__codelineno-0-13" href="#__codelineno-0-13"></a>
</span><span id="__span-0-14"><a id="__codelineno-0-14" name="__codelineno-0-14" href="#__codelineno-0-14"></a>Make UCSC hub? (default: no)
</span><span id="__span-0-15"><a id="__codelineno-0-15" name="__codelineno-0-15" href="#__codelineno-0-15"></a>  ├─ UCSC hub directory (default: seqnado_output/hub/)
</span><span id="__span-0-16"><a id="__codelineno-0-16" name="__codelineno-0-16" href="#__codelineno-0-16"></a>  ├─ Email address (default: $USER@example.com)
</span><span id="__span-0-17"><a id="__codelineno-0-17" name="__codelineno-0-17" href="#__codelineno-0-17"></a>  ├─ Genome name for hub (default: genome name from config)
</span><span id="__span-0-18"><a id="__codelineno-0-18" name="__codelineno-0-18" href="#__codelineno-0-18"></a>  └─ Color by field (default: samplename)
</span></code></pre></div>
<h3 id="atac-seq-specific">ATAC-seq Specific<a class="headerlink" href="#atac-seq-specific" title="Anchor link to this section">&para;</a></h3>
<div class="language-text highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>Shift ATAC reads? (default: yes)
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>
</span><span id="__span-1-3"><a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>Call peaks? (default: yes)
</span><span id="__span-1-4"><a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a>  ├─ Peak calling method(s) — comma-separated (default: lanceotron; options: macs, seacr, lanceotron)
</span><span id="__span-1-5"><a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a>  ├─ Generate consensus counts from Design merge column? (default: no)
</span><span id="__span-1-6"><a id="__codelineno-1-6" name="__codelineno-1-6" href="#__codelineno-1-6"></a>  └─ Run motif analysis on called peaks? (default: no)
</span><span id="__span-1-7"><a id="__codelineno-1-7" name="__codelineno-1-7" href="#__codelineno-1-7"></a>      └─ Motif analysis method(s) — if yes (default: homer; options: homer, meme)
</span><span id="__span-1-8"><a id="__codelineno-1-8" name="__codelineno-1-8" href="#__codelineno-1-8"></a>
</span><span id="__span-1-9"><a id="__codelineno-1-9" name="__codelineno-1-9" href="#__codelineno-1-9"></a>Make dataset for ML? (default: no)
</span><span id="__span-1-10"><a id="__codelineno-1-10" name="__codelineno-1-10" href="#__codelineno-1-10"></a>  └─ Use regions BED file? (default: yes)
</span><span id="__span-1-11"><a id="__codelineno-1-11" name="__codelineno-1-11" href="#__codelineno-1-11"></a>      └─ Path to regions BED file (default: path/to/regions.bed)
</span><span id="__span-1-12"><a id="__codelineno-1-12" name="__codelineno-1-12" href="#__codelineno-1-12"></a>         OR Binsize for dataset (default: 1000)
</span></code></pre></div>
<h3 id="chip-seq-specific">ChIP-seq Specific<a class="headerlink" href="#chip-seq-specific" title="Anchor link to this section">&para;</a></h3>
<div class="language-text highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>Call peaks? (default: yes)
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>  ├─ Peak calling method(s) — comma-separated (default: lanceotron; options: macs, seacr, lanceotron)
</span><span id="__span-2-3"><a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>  ├─ Generate consensus counts from Design merge column? (default: no)
</span><span id="__span-2-4"><a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a>  └─ Run motif analysis on called peaks? (default: no)
</span><span id="__span-2-5"><a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a>      └─ Motif analysis method(s) — if yes (default: homer)
</span><span id="__span-2-6"><a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a>
</span><span id="__span-2-7"><a id="__codelineno-2-7" name="__codelineno-2-7" href="#__codelineno-2-7"></a>Do you have spike-in? (default: no)
</span><span id="__span-2-8"><a id="__codelineno-2-8" name="__codelineno-2-8" href="#__codelineno-2-8"></a>  ├─ Normalisation method(s) — comma-separated (default: orlando; options: orlando, with-input, deseq2, edger, csaw)
</span><span id="__span-2-9"><a id="__codelineno-2-9" name="__codelineno-2-9" href="#__codelineno-2-9"></a>  ├─ Reference genome (default: hg38)
</span><span id="__span-2-10"><a id="__codelineno-2-10" name="__codelineno-2-10" href="#__codelineno-2-10"></a>  ├─ Spike-in genome (default: dm6)
</span><span id="__span-2-11"><a id="__codelineno-2-11" name="__codelineno-2-11" href="#__codelineno-2-11"></a>  └─ Spike-in control gene names — if using deseq2 or edger (default: AmpR,Cas9_3p,Cas9_5p)
</span><span id="__span-2-12"><a id="__codelineno-2-12" name="__codelineno-2-12" href="#__codelineno-2-12"></a>
</span><span id="__span-2-13"><a id="__codelineno-2-13" name="__codelineno-2-13" href="#__codelineno-2-13"></a>Make dataset for ML? (default: no)
</span><span id="__span-2-14"><a id="__codelineno-2-14" name="__codelineno-2-14" href="#__codelineno-2-14"></a>  └─ Use regions BED file? (default: yes)
</span><span id="__span-2-15"><a id="__codelineno-2-15" name="__codelineno-2-15" href="#__codelineno-2-15"></a>      └─ Path to regions BED file (default: path/to/regions.bed)
</span><span id="__span-2-16"><a id="__codelineno-2-16" name="__codelineno-2-16" href="#__codelineno-2-16"></a>         OR Binsize for dataset (default: 1000)
</span></code></pre></div>
<h3 id="cuttag-cat-specific">CUT&amp;Tag (CAT) Specific<a class="headerlink" href="#cuttag-cat-specific" title="Anchor link to this section">&para;</a></h3>
<div class="language-text highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>Shift CAT reads? (default: no)
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>
</span><span id="__span-3-3"><a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>Call peaks? (default: yes)
</span><span id="__span-3-4"><a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a>  ├─ Peak calling method(s) — comma-separated (default: seacr; options: macs, seacr, lanceotron)
</span><span id="__span-3-5"><a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a>  ├─ Generate consensus counts from Design merge column? (default: no)
</span><span id="__span-3-6"><a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a>  └─ Run motif analysis on called peaks? (default: no)
</span><span id="__span-3-7"><a id="__codelineno-3-7" name="__codelineno-3-7" href="#__codelineno-3-7"></a>      └─ Motif analysis method(s) — if yes (default: homer)
</span><span id="__span-3-8"><a id="__codelineno-3-8" name="__codelineno-3-8" href="#__codelineno-3-8"></a>
</span><span id="__span-3-9"><a id="__codelineno-3-9" name="__codelineno-3-9" href="#__codelineno-3-9"></a>Do you have spike-in? (default: no)
</span><span id="__span-3-10"><a id="__codelineno-3-10" name="__codelineno-3-10" href="#__codelineno-3-10"></a>  ├─ Normalisation method(s) — comma-separated (default: orlando)
</span><span id="__span-3-11"><a id="__codelineno-3-11" name="__codelineno-3-11" href="#__codelineno-3-11"></a>  ├─ Reference genome (default: hg38)
</span><span id="__span-3-12"><a id="__codelineno-3-12" name="__codelineno-3-12" href="#__codelineno-3-12"></a>  ├─ Spike-in genome (default: dm6)
</span><span id="__span-3-13"><a id="__codelineno-3-13" name="__codelineno-3-13" href="#__codelineno-3-13"></a>  └─ Spike-in control gene names — if using deseq2 or edger (default: AmpR,Cas9_3p,Cas9_5p)
</span><span id="__span-3-14"><a id="__codelineno-3-14" name="__codelineno-3-14" href="#__codelineno-3-14"></a>
</span><span id="__span-3-15"><a id="__codelineno-3-15" name="__codelineno-3-15" href="#__codelineno-3-15"></a>Make dataset for ML? (default: no)
</span><span id="__span-3-16"><a id="__codelineno-3-16" name="__codelineno-3-16" href="#__codelineno-3-16"></a>  └─ Use regions BED file? (default: yes)
</span><span id="__span-3-17"><a id="__codelineno-3-17" name="__codelineno-3-17" href="#__codelineno-3-17"></a>      └─ Path to regions BED file
</span><span id="__span-3-18"><a id="__codelineno-3-18" name="__codelineno-3-18" href="#__codelineno-3-18"></a>         OR Binsize for dataset
</span></code></pre></div>
<h3 id="rna-seq-specific">RNA-seq Specific<a class="headerlink" href="#rna-seq-specific" title="Anchor link to this section">&para;</a></h3>
<div class="language-text highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a>Do you have spike-in? (default: no)
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>  ├─ Normalisation method(s) — comma-separated (default: deseq2; options: orlando, with-input, deseq2, edger, csaw)
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>  ├─ Reference genome (default: hg38)
</span><span id="__span-4-4"><a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a>  ├─ Spike-in genome (default: spikein_rna)
</span><span id="__span-4-5"><a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a>  └─ Spike-in control gene names — if using deseq2 or edger (default: AmpR,Cas9_3p,Cas9_5p)
</span><span id="__span-4-6"><a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a>
</span><span id="__span-4-7"><a id="__codelineno-4-7" name="__codelineno-4-7" href="#__codelineno-4-7"></a>Quantification method? (default: feature_counts; options: feature_counts, salmon)
</span><span id="__span-4-8"><a id="__codelineno-4-8" name="__codelineno-4-8" href="#__codelineno-4-8"></a>  └─ Strandedness? — if feature_counts (default: 0; options: 0=unstranded, 1=forward, 2=reverse)
</span><span id="__span-4-9"><a id="__codelineno-4-9" name="__codelineno-4-9" href="#__codelineno-4-9"></a>
</span><span id="__span-4-10"><a id="__codelineno-4-10" name="__codelineno-4-10" href="#__codelineno-4-10"></a>Salmon index path? — if salmon method (default: path/to/salmon_index)
</span><span id="__span-4-11"><a id="__codelineno-4-11" name="__codelineno-4-11" href="#__codelineno-4-11"></a>
</span><span id="__span-4-12"><a id="__codelineno-4-12" name="__codelineno-4-12" href="#__codelineno-4-12"></a>Run DESeq2? (default: no)
</span></code></pre></div>
<h3 id="mcc-capture-c-specific">MCC (Capture-C) Specific<a class="headerlink" href="#mcc-capture-c-specific" title="Anchor link to this section">&para;</a></h3>
<div class="language-text highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>Call peaks? (default: yes)
</span><span id="__span-5-2"><a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a>  ├─ Peak calling method(s) — comma-separated (default: lanceotronmcc; options: macs, seacr, lanceotron, lanceotronmcc)
</span><span id="__span-5-3"><a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a>  ├─ Generate consensus counts from Design merge column? (default: no)
</span><span id="__span-5-4"><a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a>  └─ Run motif analysis on called peaks? (default: no)
</span><span id="__span-5-5"><a id="__codelineno-5-5" name="__codelineno-5-5" href="#__codelineno-5-5"></a>      └─ Motif analysis method(s) — if yes (default: homer)
</span><span id="__span-5-6"><a id="__codelineno-5-6" name="__codelineno-5-6" href="#__codelineno-5-6"></a>
</span><span id="__span-5-7"><a id="__codelineno-5-7" name="__codelineno-5-7" href="#__codelineno-5-7"></a>Path to viewpoints file (required; default: path/to/viewpoints.bed)
</span><span id="__span-5-8"><a id="__codelineno-5-8" name="__codelineno-5-8" href="#__codelineno-5-8"></a>
</span><span id="__span-5-9"><a id="__codelineno-5-9" name="__codelineno-5-9" href="#__codelineno-5-9"></a>Resolutions for MCC cooler files — comma-separated (default: 100,1000)
</span></code></pre></div>
<h3 id="methylation-specific">Methylation Specific<a class="headerlink" href="#methylation-specific" title="Anchor link to this section">&para;</a></h3>
<div class="language-text highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>Call methylation? (default: no)
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>  └─ Spike-in genomes — comma-separated, if yes (default: Lambda,250bp-v1,2kb-unmod)
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a>Methylation assay? (default: taps; options: taps, bsseq)
</span></code></pre></div>
<h3 id="snp-calling-specific">SNP Calling Specific<a class="headerlink" href="#snp-calling-specific" title="Anchor link to this section">&para;</a></h3>
<div class="language-text highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>Call SNPs? (default: yes)
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>  ├─ SNP calling method? (default: bcftools; options: bcftools)
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>  └─ Annotate SNPs? (default: no)
</span><span id="__span-7-4"><a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>      └─ Path to SNP database — if yes (default: path/to/snp_database)
</span><span id="__span-7-5"><a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>
</span><span id="__span-7-6"><a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a>Generate GEO submission files? (default: no)
</span></code></pre></div>
<h3 id="crispr-specific">CRISPR Specific<a class="headerlink" href="#crispr-specific" title="Anchor link to this section">&para;</a></h3>
<div class="language-text highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>Use MAGeCK for guide RNA analysis? (default: no)
</span><span id="__span-8-2"><a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>
</span><span id="__span-8-3"><a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a>Generate GEO submission files? (default: no)
</span></code></pre></div>
<hr />
<h3 id="understanding-multi-select-questions">Understanding Multi-Select Questions<a class="headerlink" href="#understanding-multi-select-questions" title="Anchor link to this section">&para;</a></h3>
<p>Some questions allow <strong>comma-separated selections</strong> for multiple values:
- <strong>Peak calling method(s)</strong>: Can use MACS, SEACR, and Lanceotron simultaneously
- <strong>Normalisation method(s)</strong>: Can run Orlando and DESeq2 on the same data
- <strong>Motif analysis method(s)</strong>: Can run both Homer and MEME on peaks
- <strong>Bigwig method(s)</strong>: Can generate with both deeptools and bamnado</p>
<p>Example:
<div class="language-text highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>Peak calling method(s) (comma-separated for multiple): macs, seacr, lanceotron
</span></code></pre></div></p>
<p>The pipeline will run all specified methods and generate outputs for each.</p>
<hr />
<h3 id="resource-constraints-to-consider">Resource Constraints to Consider<a class="headerlink" href="#resource-constraints-to-consider" title="Anchor link to this section">&para;</a></h3>
<p>Before finalizing your configuration, review your compute environment:</p>
<ul>
<li><strong>Memory per sample</strong>: Does your cluster have enough RAM for large BAM files (deeptools, Salmon, variant calling)?</li>
<li><strong>Runtime limits</strong>: Do you have wall-clock time limits? Some tools (STAR, MACS2, DESeq2) can be slow on large files.</li>
<li><strong>Storage</strong>: Will intermediate files (unsorted BAMs, uncompressed bigwigs) fit on your filesystem?</li>
<li><strong>Parallelization</strong>: How many samples can you process in parallel without overwhelming resources?</li>
</ul>
<p>See the <a href="../troubleshooting/#resource-allocation">Troubleshooting guide</a> for guidance on resource requirements per tool.</p>
<hr />
<h2 id="running-seqnado-config">Running <code>seqnado config</code><a class="headerlink" href="#running-seqnado-config" title="Anchor link to this section">&para;</a></h2>
<p>The <code>seqnado config</code> command interactively builds a YAML configuration file for your selected assay. It asks questions appropriate to your assay type and generates a configuration that you can review and customize before running the pipeline.</p>
<h3 id="assay-types">Assay Types<a class="headerlink" href="#assay-types" title="Anchor link to this section">&para;</a></h3>
<table>
<thead>
<tr>
<th>Assay</th>
<th>CLI name</th>
<th>When to Use</th>
</tr>
</thead>
<tbody>
<tr>
<td>ATAC-seq</td>
<td><code>atac</code></td>
<td>Open chromatin profiling (Tn5-based)</td>
</tr>
<tr>
<td>ChIP-seq</td>
<td><code>chip</code></td>
<td>Chromatin immunoprecipitation</td>
</tr>
<tr>
<td>CRISPR analysis</td>
<td><code>crispr</code></td>
<td>CRISPR screen analysis</td>
</tr>
<tr>
<td>CUT&amp;Tag</td>
<td><code>cat</code></td>
<td>Chromatin profiling by tagmentation (low background)</td>
</tr>
<tr>
<td>MCC</td>
<td><code>mcc</code></td>
<td>Capture-C or similar 3C-derived methods</td>
</tr>
<tr>
<td>Methylation</td>
<td><code>meth</code></td>
<td>Bisulfite or TAPS methylation calling</td>
</tr>
<tr>
<td>RNA-seq</td>
<td><code>rna</code></td>
<td>Quantification from RNA-seq reads</td>
</tr>
<tr>
<td>SNP analysis</td>
<td><code>snp</code></td>
<td>Variant calling and annotation</td>
</tr>
</tbody>
</table>
<p>For all available arguments and flags, see: <a href="../cli/#cli-seqnado-config">seqnado config</a>.</p>
<h3 id="example-usage">Example Usage<a class="headerlink" href="#example-usage" title="Anchor link to this section">&para;</a></h3>
<h4 id="build-a-configuration-for-chip-seq">Build a Configuration for ChIP-seq<a class="headerlink" href="#build-a-configuration-for-chip-seq" title="Anchor link to this section">&para;</a></h4>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a>seqnado<span class="w"> </span>config<span class="w"> </span>chip<span class="w"> </span>--output<span class="w"> </span>chip_config.yaml
</span></code></pre></div>
<h4 id="interactive-multiomics-configuration-multiple-assays">Interactive Multiomics Configuration (Multiple Assays)<a class="headerlink" href="#interactive-multiomics-configuration-multiple-assays" title="Anchor link to this section">&para;</a></h4>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>seqnado<span class="w"> </span>config<span class="w"> </span>--make-dirs<span class="w"> </span>--interactive
</span></code></pre></div>
<h2 id="third-party-tools">Third Party Tools<a class="headerlink" href="#third-party-tools" title="Anchor link to this section">&para;</a></h2>
<p>SeqNado integrates with specialized tools for each analysis step. Before running the pipeline, review the guidance below for tools critical to your assay. <strong>Default parameters are sensible for most experiments</strong>, but customization may be necessary for specific data, assays, and experimental designs.</p>
<h3 id="supported-tools-reference">Supported Tools Reference<a class="headerlink" href="#supported-tools-reference" title="Anchor link to this section">&para;</a></h3>
<p>Below is a comprehensive list of tools integrated into SeqNado, organized by function. <strong>⚠️ Key configuration warnings are included for critical tools.</strong></p>
<h4 id="alignment-indexing">Alignment &amp; Indexing<a class="headerlink" href="#alignment-indexing" title="Anchor link to this section">&para;</a></h4>
<ul>
<li><strong>bowtie2</strong> — Maps sequencing reads to a reference genome (ATAC, ChIP, CUT&amp;Tag, SNP)</li>
<li><strong>STAR</strong> — Splice-aware aligner for RNA-seq data (RNA)</li>
<li><strong>salmon</strong> — Pseudoalignment-based RNA quantification (RNA; optional alternative to alignment)</li>
<li>⚠️ <strong>Requires compatible reference index</strong>: GTF version must match your genome build (e.g., GRCh38 GTF for hg38)</li>
<li>Faster than alignment-based methods but less flexible for debugging strandedness</li>
<li>Generate a Salmon index before running: <code>salmon index -t transcripts.fa -i salmon_index_hg38</code></li>
</ul>
<h4 id="read-processing">Read Processing<a class="headerlink" href="#read-processing" title="Anchor link to this section">&para;</a></h4>
<ul>
<li><strong>samtools</strong> — BAM/SAM file manipulation and sorting</li>
<li><strong>picard</strong> — Java tools for high-throughput sequencing data manipulation</li>
<li><strong>cutadapt</strong> — Removes adapter sequences from reads</li>
<li><strong>trim-galore</strong> — Wrapper for Cutadapt and FastQC for quality control</li>
</ul>
<h4 id="peak-calling-chip-seq-atac-seq-cuttag-mcc">Peak Calling (ChIP-seq, ATAC-seq, CUT&amp;Tag, MCC)<a class="headerlink" href="#peak-calling-chip-seq-atac-seq-cuttag-mcc" title="Anchor link to this section">&para;</a></h4>
<ul>
<li><strong>MACS2</strong> — Model-based peak calling</li>
<li>
<p>⚠️ <strong>Peak calling mode depends on your mark</strong>:</p>
<ul>
<li><strong>Narrow mode</strong> (default) for transcription factors and sharp marks (H3K4me3, H3K27ac)</li>
<li><strong>Broad mode</strong> for diffuse marks (H3K27me3, H3K36me3). Without <code>--broad</code>, you'll miss 90% of broad domains</li>
<li>For weak ChIP-seq with few peaks, consider switching to SEACR or Lanceotron instead</li>
</ul>
</li>
<li>
<p><strong>SEACR</strong> — Peak caller optimized for low-background data (CUT&amp;Tag)</p>
</li>
<li>
<p>Configure stringency based on expected noise:</p>
<ul>
<li><code>stringent</code> mode (recommended for CUT&amp;Tag) reduces false positives</li>
<li><code>relaxed</code> mode if you're missing expected peaks</li>
<li>Threshold values control sensitivity vs. specificity trade-off</li>
</ul>
</li>
<li>
<p><strong>Lanceotron</strong> — Deep learning–based peak caller for broad and narrow peaks</p>
</li>
<li>No hyper-parameter tuning needed; model is pre-trained</li>
<li>
<p>Can be slow on large files; consider testing on a sample first</p>
</li>
<li>
<p><strong>lanceotronmcc</strong> — Specialized Lanceotron variant for MCC interaction peaks</p>
</li>
</ul>
<h4 id="quantification-rna-seq">Quantification (RNA-seq)<a class="headerlink" href="#quantification-rna-seq" title="Anchor link to this section">&para;</a></h4>
<ul>
<li><strong>featureCounts</strong> (subread package) — Assigns aligned reads to genomic features for RNA-seq quantification</li>
<li>
<p>⚠️ <strong>Critical GTF and strandedness configuration</strong>:</p>
<ul>
<li>Default parameters count reads at the <strong>exon level</strong> (<code>-t exon</code>) and aggregate by <strong>gene_id</strong></li>
<li><strong>Must verify GTF attribute match</strong>: Does your GTF use <code>gene_id</code> or <code>gene_name</code>? featureCounts requires exact match or all counts will be zero</li>
<li><strong>Strandedness must match library prep</strong>: Use <code>0</code> (unstranded), <code>1</code> (forward), or <code>2</code> (reverse). Using wrong strandedness results in 50–90% fewer counts</li>
<li>Paired-end mode (<code>-p --countReadPairs</code>) is auto-detected from FASTQ files</li>
<li>If all genes get zero counts, check GTF feature attribute match first</li>
</ul>
</li>
<li>
<p><strong>Salmon</strong> — Faster pseudoalignment-based quantification (requires compatible index; see Alignment &amp; Indexing above)</p>
</li>
</ul>
<h4 id="bigwig-generation-visualization">Bigwig Generation &amp; Visualization<a class="headerlink" href="#bigwig-generation-visualization" title="Anchor link to this section">&para;</a></h4>
<ul>
<li><strong>deeptools bamCoverage</strong> — Generates BigWig files from BAM alignments</li>
<li>**Using deeptools' native <code>--normalizeUsing</code> ** — To use deeptools' built-in normalization methods (<code>RPKM</code>, <code>CPM</code>, <code>BPM</code>, <code>RPGC</code>), select <code>unscaled</code> for your bigwig scaling method, then manually edit <code>third_party_tools.deeptools.bam_coverage.command_line_arguments</code> to add <code>--normalizeUsing</code> (e.g., <code>"--binSize 10 --normalizeUsing RPKM"</code>). This disables SeqNado's scaling entirely. Example use case: comparing standard RPKM-normalized bigwigs across studies</li>
<li><strong>Bigwig scaling method</strong> — You choose during config (options: <code>unscaled</code>, <code>csaw</code>):<ul>
<li><strong><code>unscaled</code></strong>: Raw read coverage without normalization; useful for visual inspection and as input for downstream analysis tools</li>
<li><strong><code>csaw</code></strong> (ChIP-Seq Analysis with Windows): Applies scaling factors between samples to equalize library depth while preserving broad features; recommended when comparing ChIP-seq samples or using spike-in controls; see <a href="../normalisation/">Normalisation Methods</a></li>
</ul>
</li>
<li><strong>Spike-in normalisation for bigwigs</strong> — If you selected spike-in normalisation (Orlando, with-input, DESeq2, or edgeR) during config, scale factors are automatically calculated and applied to bigwig generation, producing spike-in–normalized bigwigs for downstream analysis; see <a href="../normalisation/">Normalisation Methods</a></li>
</ul>
<div class="admonition info">
<p class="admonition-title">Info</p>
<p><strong>How SeqNado prevents conflicts between <code>--normalizeUsing</code> and <code>--scaleFactor</code></strong>: The default deeptools config includes <code>--normalizeUsing RPKM</code>. However, when you select spike-in or csaw normalization, SeqNado <strong>automatically removes</strong> <code>--normalizeUsing</code> and applies <strong>only</strong> <code>--scaleFactor</code>. This prevents deeptools from silently ignoring your scale factors (since <code>--normalizeUsing</code> overrides <code>--scaleFactor</code> in deeptools). Result: your spike-in/csaw scale factors apply correctly without unwanted RPKM normalization</p>
</div>
<ul>
<li>
<p><strong>bamnado</strong> — Alternative tool for BigWig generation and BAM manipulation</p>
</li>
<li>
<p><strong>deeptools heatmap</strong> — Creates heatmaps from BigWig files around genomic coordinates</p>
</li>
<li>
<p><strong>deeptools metaplot</strong> — Generates metaplots of signal around features</p>
</li>
</ul>
<h4 id="motif-analysis-peak-regions">Motif Analysis (peak regions)<a class="headerlink" href="#motif-analysis-peak-regions" title="Anchor link to this section">&para;</a></h4>
<ul>
<li><strong>Homer</strong> — Motif discovery and annotation in peak regions</li>
<li><strong>MEME</strong> — Alternative motif discovery tool</li>
</ul>
<h4 id="variant-calling-snp">Variant Calling (SNP)<a class="headerlink" href="#variant-calling-snp" title="Anchor link to this section">&para;</a></h4>
<ul>
<li><strong>bcftools</strong> — SNP calling and VCF manipulation</li>
<li><strong>SnpEff/SnpSift</strong> — SNP annotation and variant functional impact prediction</li>
</ul>
<h4 id="methylation-specialized">Methylation &amp; Specialized<a class="headerlink" href="#methylation-specialized" title="Anchor link to this section">&para;</a></h4>
<ul>
<li><strong>methyldackel</strong> — Extract methylation calls from bisulfite-seq data</li>
<li><strong>MAGeCK</strong> — Statistical analysis of CRISPR screen data</li>
<li><strong>UCSC utilities</strong> — Convert and manage genome tracks in UCSC format</li>
</ul>
<h4 id="data-analysis-normalisation">Data Analysis &amp; Normalisation<a class="headerlink" href="#data-analysis-normalisation" title="Anchor link to this section">&para;</a></h4>
<ul>
<li>
<p><strong>DESeq2</strong> (R/Bioconductor) — Differential expression analysis (RNA-seq)</p>
</li>
<li>
<p><strong>edgeR</strong> (R/Bioconductor) — Differential expression with spike-in normalisation</p>
</li>
<li>
<p>⚠️ <strong>Spike-in normalisation setup</strong>:</p>
<ul>
<li>Requires correct spike-in genome specification (e.g., <code>dm6</code> for <em>Drosophila</em>)</li>
<li>Control gene names must match spike-in reference GTF (e.g., <code>ERCC</code>)</li>
<li>With-input normalisation requires ChIP and input samples paired in design file — see <a href="../design/">Design guide</a></li>
</ul>
</li>
<li>
<p><strong>Orlando</strong> — Spike-in normalisation method for chromatin-based assays (ChIP, CAT)</p>
</li>
<li>
<p>Recommended for spike-in data with balanced endogenous and exogenous reads</p>
</li>
<li>
<p><strong>with-input</strong> — Normalisation using paired input controls (ChIP-seq only)</p>
</li>
<li>Requires input samples correctly paired in design file</li>
<li>
<p>Alternative to spike-in normalisation when spike-ins unavailable</p>
</li>
<li>
<p><strong>CSAW</strong> — Cyclic shift aware normalisation; produces merged bigwigs</p>
</li>
<li>Recommended for spike-in data with unbalanced read distributions</li>
<li>See <a href="../normalisation/">Normalisation Methods</a> for detailed comparison</li>
</ul>
<h3 id="configuration-logic-tool-interactions">Configuration Logic &amp; Tool Interactions<a class="headerlink" href="#configuration-logic-tool-interactions" title="Anchor link to this section">&para;</a></h3>
<p>See the <a href="../pipeline/#supported-assays">Pipeline Overview</a> for guidance on which tools to use for each assay type.</p>
<p>For detailed normalisation methods comparison, see <a href="../normalisation/">Normalisation Methods</a>.</p>
<p>For configuration command options and usage patterns, see <a href="../cli/#cli-seqnado-config">seqnado config</a>.</p>
<hr />
<h2 id="common-pitfalls-and-best-practices">Common Pitfalls and Best Practices<a class="headerlink" href="#common-pitfalls-and-best-practices" title="Anchor link to this section">&para;</a></h2>
<h3 id="pitfalls-to-avoid">Pitfalls to Avoid<a class="headerlink" href="#pitfalls-to-avoid" title="Anchor link to this section">&para;</a></h3>
<p><strong>Assuming defaults are correct without validation</strong> 
   - <strong>Problem</strong>: Accepting default peak caller (lanceotron), GTF attributes, or strandedness without checking your specific data
   - <strong>Fix</strong>: For each new assay/GTF/strandedness combo, test peak calling or quantification on 1–2 samples first; spot-check BAM files and counts</p>
<p><strong>Mismatched GTF and genome builds</strong>
   - <strong>Problem</strong>: Using GRCh37 GTF with hg38 genome, or mm10 GTF with mm39
   - <strong>Fix</strong>: Verify your annotation version matches your genome build at download time; document version numbers in your README</p>
<p><strong>Wrong peak calling mode for your biological mark</strong>
   - <strong>Problem</strong>: Using MACS2 narrow mode for H3K27me3 (broad mark) → misses 90% of signal; using broad mode for H3K4me3 (sharp mark) → thousands of false positives
   - <strong>Fix</strong>: Know your mark. H3K4me3, H3K27ac, TF binding → narrow. H3K27me3, H3K36me3, H3K9me3 → broad. Test on a rep.</p>
<p><strong>Forgetting to pair ChIP and input samples in design file</strong>
   - <strong>Problem</strong>: Configuring "with-input" normalisation but not ensuring control sample pairs in <a href="../design/">Design</a>
   - <strong>Fix</strong>: Before running pipeline, review design file to confirm ChIP→input pairing syntax matched for all samples</p>
<p><strong>Ignoring resource constraints</strong>
   - <strong>Problem</strong>: Setting memory-intensive parameters (STAR <code>--limitSjdbInsertNsj</code>, deeptools multi-threaded) on systems without available RAM
   - <strong>Fix</strong>: Test on 1 sample and monitor resource usage (<code>top</code>, cluster monitoring); adjust thread count or binsize if needed</p>
<h3 id="best-practices">Best Practices<a class="headerlink" href="#best-practices" title="Anchor link to this section">&para;</a></h3>
<p>✅ <strong>Test before large-scale runs</strong>
- Run <code>seqnado snakemake ... -n</code> (dry-run) to visualize the workflow DAG before execution
- Test peak calling or quantification on 1–2 samples to validate your tool selections
- Spot-check intermediate outputs: BAM file headers (correct chromosome names?), peak files (reasonable count?), count matrices (non-zero?), BigWigs (visible in IGV?)</p>
<p>✅ <strong>Version-control your configuration files</strong>
- Commit config YAML files to git so analysis is fully reproducible
- Create assay-specific configs (e.g., <code>config_chip_h3k27ac.yaml</code>, <code>config_rna_pe150.yaml</code>) so you can reuse templates
- Different analyses can and should use different configs — this is your version control</p>
<p>✅ <strong>Validate reference files upfront</strong>
- Before running the full pipeline, verify:
  - GTF file: Spot-check a few known genes and their feature definitions
  - Spike-in genome: Confirm the spike-in species (dm6? ecoli?) is included in your genome build
  - Salmon index: If using Salmon, ensure the index matches your GTF and genome version
  - bedtools/peaks BED files: Verify chromosome names match your genome (e.g., <code>chr1</code> vs <code>1</code>)</p>
<p>✅ <strong>Monitor the first few samples carefully</strong>
- Check the first sample's outputs in detail before running the full cohort
- Verify BAM file alignment rate (expect &gt;90% for most data)
- Check peak files are reasonable (not zero peaks, not millions of false positives)
- Confirm BigWig tracks look correct in IGV (visible signal? expected chromosome coverage?)
- This catches configuration errors early and saves days of compute time</p>
<p>✅ <strong>Use meaningful output names</strong>
- Use the <code>--output</code> flag to give your config file a descriptive name: <code>seqnado config rna --output lps_response_config.yaml</code>
- This helps keep track of different analyses without confusion</p>
<hr />
<p><strong>See Also:</strong></p>
<ul>
<li><a href="../design/">Design Guide</a> - Create experimental design files</li>
<li><a href="../tools/">Tools Reference</a> - Configure tool-specific options</li>
<li><a href="../troubleshooting/#configuration-seqnado-config">Troubleshooting</a> - Configuration issues</li>
</ul>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../genomes/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Genome Setup">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Genome Setup
              </div>
            </div>
          </a>
        
        
          
          <a href="../design/" class="md-footer__link md-footer__link--next" aria-label="Next: Design Guide">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                Design Guide
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2024 - 2026 Milne Group
    </div>
  
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
    <a href="https://github.com/Milne-Group/SeqNado" target="_blank" rel="noopener" title="SeqNado on GitHub" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
    
    
    
    
    <a href="https://pypi.org/project/seqnado" target="_blank" rel="noopener" title="SeqNado on PyPI" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.8 200.5c-7.7-30.9-22.3-54.2-53.4-54.2h-40.1v47.4c0 36.8-31.2 67.8-66.8 67.8H172.7c-29.2 0-53.4 25-53.4 54.3v101.8c0 29 25.2 46 53.4 54.3 33.8 9.9 66.3 11.7 106.8 0 26.9-7.8 53.4-23.5 53.4-54.3v-40.7H226.2v-13.6h160.2c31.1 0 42.6-21.7 53.4-54.2 11.2-33.5 10.7-65.7 0-108.6M286.2 444.7a20.4 20.4 0 1 1 0-40.7 20.4 20.4 0 1 1 0 40.7M167.8 248.1h106.8c29.7 0 53.4-24.5 53.4-54.3V91.9c0-29-24.4-50.7-53.4-55.6-35.8-5.9-74.7-5.6-106.8.1-45.2 8-53.4 24.7-53.4 55.6v40.7h106.9v13.6h-147c-31.1 0-58.3 18.7-66.8 54.2-9.8 40.7-10.2 66.1 0 108.6 7.6 31.6 25.7 54.2 56.8 54.2H101v-48.8c0-35.3 30.5-66.4 66.8-66.4m-6.6-183.4a20.4 20.4 0 1 1 0 40.8 20.4 20.4 0 1 1 0-40.8"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
      <div class="md-progress" data-md-component="progress" role="progressbar"></div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "..", "features": ["announce.dismiss", "content.action.edit", "content.action.view", "content.code.annotate", "content.code.copy", "content.tabs.link", "content.tooltips", "header.autohide", "navigation.expand", "navigation.footer", "navigation.indexes", "navigation.instant", "navigation.instant.prefetch", "navigation.instant.progress", "navigation.path", "navigation.sections", "navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "toc.follow"], "search": "../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script>
    
    
      <script src="../assets/javascripts/bundle.79ae519e.min.js"></script>
      
        <script src="../javascripts/toc-collapse.js"></script>
      
        <script src="../javascripts/mathjax.js"></script>
      
        <script src="https://unpkg.com/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>