
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="A Snakemake-based bioinformatics toolkit for analyzing sequencing data">
      
      
        <meta name="author" content="Milne Group">
      
      
        <link rel="canonical" href="https://milne-group.github.io/SeqNado/outputs/">
      
      
        <link rel="prev" href="../pipeline/">
      
      
        <link rel="next" href="../examples/">
      
      
        
      
      
      <link rel="icon" href="https://raw.githubusercontent.com/Milne-Group/SeqNado/main/containers/pipeline/seqnado.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.2">
    
    
      
        <title>Outputs - SeqNado</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.484c7ddc.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../assets/_mkdocstrings.css">
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="green" data-md-color-accent="light-green">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#pipeline-outputs" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="SeqNado" class="md-header__button md-logo" aria-label="SeqNado" data-md-component="logo">
      
  <img src="https://raw.githubusercontent.com/Milne-Group/SeqNado/main/containers/pipeline/seqnado.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            SeqNado
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Outputs
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="green" data-md-color-accent="light-green"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5s-1.65.15-2.39.42zM3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29zm.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14zM20.65 7l-1.77 3.79a7.02 7.02 0 0 0-2.38-4.15zm-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29zM12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="teal" data-md-color-accent="lime"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3zm3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95zm-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/Milne-Group/SeqNado" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    Milne-Group/SeqNado
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href=".." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../quick_start/" class="md-tabs__link">
          
  
  
    
  
  Getting Started

        </a>
      </li>
    
  

      
        
  
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../configuration/" class="md-tabs__link">
          
  
  
    
  
  Usage

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../api/" class="md-tabs__link">
          
  
  
    
  
  Reference

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../troubleshooting/" class="md-tabs__link">
          
  
  
    
  
  Help

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="SeqNado" class="md-nav__button md-logo" aria-label="SeqNado" data-md-component="logo">
      
  <img src="https://raw.githubusercontent.com/Milne-Group/SeqNado/main/containers/pipeline/seqnado.png" alt="logo">

    </a>
    SeqNado
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/Milne-Group/SeqNado" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
  </div>
  <div class="md-source__repository">
    Milne-Group/SeqNado
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Getting Started
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Getting Started
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../quick_start/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Quick Start
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../installation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Installation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../initialisation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Initialisation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../genomes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Genome Setup
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
        
        
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" checked>
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Usage
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Usage
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../configuration/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Configuration
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../design/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Design Guide
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../geo_download/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GEO/SRA Download
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pipeline/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Pipeline Overview
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Outputs
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Outputs
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#general-output-structure" class="md-nav__link">
    <span class="md-ellipsis">
      
        General Output Structure
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#main-entry-point-seqnado-report" class="md-nav__link">
    <span class="md-ellipsis">
      
        Main Entry Point: SeqNado Report
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#core-output-files" class="md-nav__link">
    <span class="md-ellipsis">
      
        Core Output Files
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Core Output Files">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#alignment-files-aligned" class="md-nav__link">
    <span class="md-ellipsis">
      
        Alignment Files (aligned/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#coverage-tracks-bigwigs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Coverage Tracks (bigwigs/)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Coverage Tracks (bigwigs/)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#normalisation-factor-calculation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Normalisation factor calculation
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#peak-calls-peaks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Peak Calls (peaks/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#read-counts-readcounts" class="md-nav__link">
    <span class="md-ellipsis">
      
        Read Counts (readcounts/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-control-qc" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quality Control (qc/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ucsc-genome-browser-hub-hub" class="md-nav__link">
    <span class="md-ellipsis">
      
        UCSC Genome Browser Hub (hub/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#heatmaps-heatmap" class="md-nav__link">
    <span class="md-ellipsis">
      
        Heatmaps (heatmap/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#motif-analysis-motifs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Motif Analysis (motifs/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-browser-plots-genome_browser_plots" class="md-nav__link">
    <span class="md-ellipsis">
      
        Genome Browser Plots (genome_browser_plots/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#geo-submission-geo_submission" class="md-nav__link">
    <span class="md-ellipsis">
      
        GEO Submission (geo_submission/)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assay-specific-outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assay-Specific Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Assay-Specific Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#atac-seq" class="md-nav__link">
    <span class="md-ellipsis">
      
        ATAC-seq
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#chip-seq" class="md-nav__link">
    <span class="md-ellipsis">
      
        ChIP-seq
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#cuttag" class="md-nav__link">
    <span class="md-ellipsis">
      
        CUT&amp;Tag
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#rna-seq" class="md-nav__link">
    <span class="md-ellipsis">
      
        RNA-seq
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#methylation-meth" class="md-nav__link">
    <span class="md-ellipsis">
      
        Methylation (METH)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#snp-variant-calling" class="md-nav__link">
    <span class="md-ellipsis">
      
        SNP / Variant Calling
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#micro-capture-c-mcc" class="md-nav__link">
    <span class="md-ellipsis">
      
        Micro Capture-C (MCC)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#crispr-screens" class="md-nav__link">
    <span class="md-ellipsis">
      
        CRISPR Screens
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#accessing-your-results" class="md-nav__link">
    <span class="md-ellipsis">
      
        Accessing Your Results
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Accessing Your Results">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#command-line" class="md-nav__link">
    <span class="md-ellipsis">
      
        Command Line
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#opening-reports" class="md-nav__link">
    <span class="md-ellipsis">
      
        Opening Reports
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#finding-specific-outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Finding Specific Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Finding Specific Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#peak-calling-results" class="md-nav__link">
    <span class="md-ellipsis">
      
        Peak calling results
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#coverage-tracks-for-visualisation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Coverage tracks for visualisation
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qc-html-reports" class="md-nav__link">
    <span class="md-ellipsis">
      
        QC HTML reports
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Output Examples
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../normalisation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Normalisation Methods
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tools/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Third-Party Tools
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../cli/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    CLI Reference
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_4" >
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../api/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Reference
  

    
  </span>
  
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_4" id="__nav_4_label" tabindex="0">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Reference
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/core/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Core Types
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/config/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Configuration API
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/inputs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Inputs API
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../api/outputs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Outputs API
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../assays/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Assay Configurations
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tool-config-examples/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Tool Config Examples
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../citation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Citation Guidelines
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
    
    
    
      
      
        
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Help
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Help
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../troubleshooting/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Troubleshooting
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../cluster_config/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    HPC Clusters
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../faq/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    FAQ
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#general-output-structure" class="md-nav__link">
    <span class="md-ellipsis">
      
        General Output Structure
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#main-entry-point-seqnado-report" class="md-nav__link">
    <span class="md-ellipsis">
      
        Main Entry Point: SeqNado Report
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#core-output-files" class="md-nav__link">
    <span class="md-ellipsis">
      
        Core Output Files
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Core Output Files">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#alignment-files-aligned" class="md-nav__link">
    <span class="md-ellipsis">
      
        Alignment Files (aligned/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#coverage-tracks-bigwigs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Coverage Tracks (bigwigs/)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Coverage Tracks (bigwigs/)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#normalisation-factor-calculation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Normalisation factor calculation
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#peak-calls-peaks" class="md-nav__link">
    <span class="md-ellipsis">
      
        Peak Calls (peaks/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#read-counts-readcounts" class="md-nav__link">
    <span class="md-ellipsis">
      
        Read Counts (readcounts/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-control-qc" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quality Control (qc/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ucsc-genome-browser-hub-hub" class="md-nav__link">
    <span class="md-ellipsis">
      
        UCSC Genome Browser Hub (hub/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#heatmaps-heatmap" class="md-nav__link">
    <span class="md-ellipsis">
      
        Heatmaps (heatmap/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#motif-analysis-motifs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Motif Analysis (motifs/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genome-browser-plots-genome_browser_plots" class="md-nav__link">
    <span class="md-ellipsis">
      
        Genome Browser Plots (genome_browser_plots/)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#geo-submission-geo_submission" class="md-nav__link">
    <span class="md-ellipsis">
      
        GEO Submission (geo_submission/)
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assay-specific-outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Assay-Specific Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Assay-Specific Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#atac-seq" class="md-nav__link">
    <span class="md-ellipsis">
      
        ATAC-seq
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#chip-seq" class="md-nav__link">
    <span class="md-ellipsis">
      
        ChIP-seq
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#cuttag" class="md-nav__link">
    <span class="md-ellipsis">
      
        CUT&amp;Tag
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#rna-seq" class="md-nav__link">
    <span class="md-ellipsis">
      
        RNA-seq
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#methylation-meth" class="md-nav__link">
    <span class="md-ellipsis">
      
        Methylation (METH)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#snp-variant-calling" class="md-nav__link">
    <span class="md-ellipsis">
      
        SNP / Variant Calling
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#micro-capture-c-mcc" class="md-nav__link">
    <span class="md-ellipsis">
      
        Micro Capture-C (MCC)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#crispr-screens" class="md-nav__link">
    <span class="md-ellipsis">
      
        CRISPR Screens
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#accessing-your-results" class="md-nav__link">
    <span class="md-ellipsis">
      
        Accessing Your Results
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Accessing Your Results">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#command-line" class="md-nav__link">
    <span class="md-ellipsis">
      
        Command Line
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#opening-reports" class="md-nav__link">
    <span class="md-ellipsis">
      
        Opening Reports
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#finding-specific-outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Finding Specific Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Finding Specific Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#peak-calling-results" class="md-nav__link">
    <span class="md-ellipsis">
      
        Peak calling results
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#coverage-tracks-for-visualisation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Coverage tracks for visualisation
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#qc-html-reports" class="md-nav__link">
    <span class="md-ellipsis">
      
        QC HTML reports
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
                



  


  <nav class="md-path" aria-label="Navigation" >
    <ol class="md-path__list">
      
        
  
  
    <li class="md-path__item">
      <a href=".." class="md-path__link">
        
  <span class="md-ellipsis">
    Home
  </span>

      </a>
    </li>
  

      
      
        
  
  
    
    
      <li class="md-path__item">
        <a href="../configuration/" class="md-path__link">
          
  <span class="md-ellipsis">
    Usage
  </span>

        </a>
      </li>
    
  

      
    </ol>
  </nav>

              
              <article class="md-content__inner md-typeset">
                
                  


  
    <a href="https://github.com/Milne-Group/SeqNado/edit/main/docs/outputs.md" title="Edit this page" class="md-content__button md-icon" rel="edit">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75z"/></svg>
    </a>
  
  
    
      
    
    <a href="https://github.com/Milne-Group/SeqNado/raw/main/docs/outputs.md" title="View source of this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0 8a5 5 0 0 1-5-5 5 5 0 0 1 5-5 5 5 0 0 1 5 5 5 5 0 0 1-5 5m0-12.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5"/></svg>
    </a>
  


<p><a href="../">← Back to main page</a></p>
<h1 id="pipeline-outputs">Pipeline Outputs<a class="headerlink" href="#pipeline-outputs" title="Anchor link to this section">&para;</a></h1>
<p>All SeqNado analysis results are organized within the <code>seqnado_output/</code> directory (or your custom output directory specified during configuration). This page describes the structure and types of files you can expect from your pipeline runs.</p>
<h2 id="general-output-structure">General Output Structure<a class="headerlink" href="#general-output-structure" title="Anchor link to this section">&para;</a></h2>
<div class="language-text highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>seqnado_output/{assay}/       # Assay-specific directory
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>├── seqnado_report.html       # Main interactive QC report (MultiQC)
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>├── protocol.txt              # Auto-generated data processing protocol
</span><span id="__span-0-4"><a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>├── aligned/                  # Final BAM alignment files
</span><span id="__span-0-5"><a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>├── bigwigs/                  # BigWig coverage tracks
</span><span id="__span-0-6"><a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a>├── peaks/                    # Peak calling results (ATAC, ChIP, CUT&amp;Tag)
</span><span id="__span-0-7"><a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a>├── readcounts/               # Quantification files (RNA, CRISPR)
</span><span id="__span-0-8"><a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a>├── qc/                       # Quality control metrics
</span><span id="__span-0-9"><a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a>├── hub/                      # UCSC Genome Browser hub
</span><span id="__span-0-10"><a id="__codelineno-0-10" name="__codelineno-0-10" href="#__codelineno-0-10"></a>├── heatmap/                  # DeepTools heatmap and metaplot PDFs
</span><span id="__span-0-11"><a id="__codelineno-0-11" name="__codelineno-0-11" href="#__codelineno-0-11"></a>├── motifs/                   # Motif analysis results (if enabled)
</span><span id="__span-0-12"><a id="__codelineno-0-12" name="__codelineno-0-12" href="#__codelineno-0-12"></a>├── tag_dirs/                 # HOMER tag directories
</span><span id="__span-0-13"><a id="__codelineno-0-13" name="__codelineno-0-13" href="#__codelineno-0-13"></a>├── resources/                # Normalisation factors (spike-in and CSAW)
</span><span id="__span-0-14"><a id="__codelineno-0-14" name="__codelineno-0-14" href="#__codelineno-0-14"></a>├── genome_browser_plots/     # PlotNado visualisations (if configured)
</span><span id="__span-0-15"><a id="__codelineno-0-15" name="__codelineno-0-15" href="#__codelineno-0-15"></a>├── geo_submission/           # GEO submission-ready files (if enabled)
</span><span id="__span-0-16"><a id="__codelineno-0-16" name="__codelineno-0-16" href="#__codelineno-0-16"></a>├── methylation/              # Methylation calls (METH only)
</span><span id="__span-0-17"><a id="__codelineno-0-17" name="__codelineno-0-17" href="#__codelineno-0-17"></a>├── variant/                  # VCF files (SNP only)
</span><span id="__span-0-18"><a id="__codelineno-0-18" name="__codelineno-0-18" href="#__codelineno-0-18"></a>└── logs/                     # Process execution logs
</span></code></pre></div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Not all directories will be present for every assay. The exact output depends on your assay type and configuration options.</p>
</div>
<h2 id="main-entry-point-seqnado-report">Main Entry Point: SeqNado Report<a class="headerlink" href="#main-entry-point-seqnado-report" title="Anchor link to this section">&para;</a></h2>
<p>The <strong><code>seqnado_report.html</code></strong> file is your primary analysis report, generated by MultiQC. It provides:</p>
<ul>
<li><strong>QC Summary</strong>: FastQC, alignment statistics, and quality metrics</li>
<li><strong>Sample Overview</strong>: All samples displayed with key metrics</li>
<li><strong>Peak Statistics</strong>: Number of peaks called, genomic distribution (for applicable assays)</li>
<li><strong>Library Complexity</strong>: Duplication rates and unique read counts</li>
<li><strong>Multi-Sample Comparisons</strong>: Side-by-side QC metrics</li>
</ul>
<p>A <strong><code>protocol.txt</code></strong> file is also generated, providing an auto-generated data processing protocol describing the steps performed.</p>
<div class="admonition tip">
<p class="admonition-title">Viewing the Report</p>
<p>Open <code>seqnado_report.html</code> in any modern web browser. No server required!</p>
</div>
<h2 id="core-output-files">Core Output Files<a class="headerlink" href="#core-output-files" title="Anchor link to this section">&para;</a></h2>
<h3 id="alignment-files-aligned">Alignment Files (<code>aligned/</code>)<a class="headerlink" href="#alignment-files-aligned" title="Anchor link to this section">&para;</a></h3>
<p>The BAM processing pipeline runs through several intermediate stages (sorting, blacklist filtering, duplicate removal, optional Tn5 shifting, quality filtering), but <strong>all intermediate files are temporary and deleted after the pipeline completes</strong>. Only the final processed BAM files are retained:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>aligned/
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>├── {sample}.bam                  # Final processed BAM
</span><span id="__span-1-3"><a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>└── {sample}.bam.bai              # BAM index
</span></code></pre></div>
<p><strong>Processing chain</strong> (intermediates are removed automatically):</p>
<p><code>raw → sorted → blacklist_regions_removed → duplicates_removed → shifted_for_tn5_insertion (ATAC/CUT&amp;Tag only) → filtered → final</code></p>
<p><strong>File Formats:</strong></p>
<ul>
<li><strong>BAM</strong>: Binary alignment format, viewable with samtools/IGV</li>
<li><strong>BAI</strong>: Index files for rapid random access</li>
</ul>
<h3 id="coverage-tracks-bigwigs">Coverage Tracks (<code>bigwigs/</code>)<a class="headerlink" href="#coverage-tracks-bigwigs" title="Anchor link to this section">&para;</a></h3>
<p>Genome-wide signal tracks for visualisation, organised by tool and scaling method:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>bigwigs/
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>├── {method}/                              # deeptools, homer, or bamnado
</span><span id="__span-2-3"><a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>│   ├── {sample}.bigWig                    # Default unscaled tracks
</span><span id="__span-2-4"><a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a>│   ├── csaw/                              # CSAW-normalised individual (if enabled)
</span><span id="__span-2-5"><a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a>│   │   └── {sample}.bigWig
</span><span id="__span-2-6"><a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a>│   ├── spikein/                           # Spike-in normalised (if applicable)
</span><span id="__span-2-7"><a id="__codelineno-2-7" name="__codelineno-2-7" href="#__codelineno-2-7"></a>│   │   └── {spikein_method}/
</span><span id="__span-2-8"><a id="__codelineno-2-8" name="__codelineno-2-8" href="#__codelineno-2-8"></a>│   │       └── {sample}.bigWig
</span><span id="__span-2-9"><a id="__codelineno-2-9" name="__codelineno-2-9" href="#__codelineno-2-9"></a>│   └── merged/                            # Consensus group merged tracks
</span><span id="__span-2-10"><a id="__codelineno-2-10" name="__codelineno-2-10" href="#__codelineno-2-10"></a>│       ├── {group}.bigWig                 # Unscaled merged
</span><span id="__span-2-11"><a id="__codelineno-2-11" name="__codelineno-2-11" href="#__codelineno-2-11"></a>│       ├── csaw/                          # CSAW-scaled merged (if enabled)
</span><span id="__span-2-12"><a id="__codelineno-2-12" name="__codelineno-2-12" href="#__codelineno-2-12"></a>│       │   └── {group}.bigWig
</span><span id="__span-2-13"><a id="__codelineno-2-13" name="__codelineno-2-13" href="#__codelineno-2-13"></a>│       └── spikein/                       # Spike-in scaled merged (if applicable)
</span><span id="__span-2-14"><a id="__codelineno-2-14" name="__codelineno-2-14" href="#__codelineno-2-14"></a>│           └── {spikein_method}/
</span><span id="__span-2-15"><a id="__codelineno-2-15" name="__codelineno-2-15" href="#__codelineno-2-15"></a>│               └── {group}.bigWig
</span></code></pre></div>
<p>For <strong>RNA-seq</strong>, stranded bigwigs are produced with <code>_plus</code> and <code>_minus</code> suffixes:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>bigwigs/{method}/{sample}_plus.bigWig
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>bigwigs/{method}/{sample}_minus.bigWig
</span></code></pre></div>
<p><strong>Pileup Tools:</strong></p>
<ul>
<li><strong>DeepTools</strong>: <code>bamCoverage</code>-based tracks (supports all scaling methods, individual and merged)</li>
<li><strong>HOMER</strong>: <code>makeBigWig.pl</code>-based tracks (unscaled individual and unscaled merged only)</li>
<li><strong>BamNado</strong>: Custom pileup tool (unscaled and CSAW individual; unscaled and CSAW merged)</li>
</ul>
<p><strong>Scaling Methods:</strong></p>
<ul>
<li><strong>unscaled</strong>: No normalisation applied</li>
<li><strong>csaw</strong>: CSAW-based normalisation using binned read counts across samples</li>
<li><strong>spikein</strong>: Spike-in normalisation using external control DNA (DeepTools only)</li>
</ul>
<p><strong>Merged Tracks</strong> (when <code>consensus_group</code> is set in the design file):</p>
<p>Individual-sample bigwigs are complemented by merged tracks for each consensus group, supporting all of the same scaling methods as individual tracks (where applicable). CSAW and spike-in scaled merged bigwigs require the corresponding normalization to be enabled.</p>
<h4 id="normalisation-factor-calculation">Normalisation factor calculation<a class="headerlink" href="#normalisation-factor-calculation" title="Anchor link to this section">&para;</a></h4>
<p><strong>Per-sample bigwigs</strong> use a factor calculated individually for each sample. For CSAW, all samples within a scaling group are compared via binned read counts and each sample receives:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a>scale_factor = mean_library_size_of_group / sample_library_size
</span></code></pre></div>
<p>Samples with larger libraries are scaled down; samples with smaller libraries are scaled up. The results are written to <code>resources/{group}_scaling_factors.tsv</code>.</p>
<p><strong>Merged bigwigs</strong> are generated from a BAM produced by <code>samtools merge</code> (concatenating all reads in the consensus group). The scale factor applied to the merged BAM is the <strong>arithmetic mean of the per-sample factors</strong> for all samples in that group:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-5-1"><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>merged_scale_factor = mean(scale_factor_sample_1, scale_factor_sample_2, ...)
</span></code></pre></div>
<p>This ensures the merged track is normalised to the average depth of the constituent samples.</p>
<p>For <strong>spike-in normalised merged bigwigs</strong>, a simple arithmetic mean of per-sample factors would give incorrect results when samples differ in sequencing depth (deeply-sequenced samples contribute far more reads to the merged BAM but their smaller factors would be under-weighted). Instead, SeqNado derives the merged factor from the pooled spike-in counts, equivalent to "merge-then-split":</p>
<ul>
<li><strong>orlando</strong>: <code>merged_factor = 1e6 / sum(spikein_reads)</code> — exact pooled calculation using each sample's <code>aligned/spikein/{sample}_stats.tsv</code></li>
<li><strong>with_input</strong>: spike-in-weighted mean of per-sample factors, <code>sum(f_i × S_ip_i) / sum(S_ip_i)</code>, which correctly weights samples by their sequencing depth</li>
<li><strong>deseq2 / edgeR</strong>: arithmetic mean fallback (these factors come from fitted statistical models rather than raw spike-in counts)</li>
</ul>
<table>
<thead>
<tr>
<th></th>
<th>Per-sample bigwig</th>
<th>Merged bigwig</th>
</tr>
</thead>
<tbody>
<tr>
<td>Input BAM</td>
<td><code>aligned/{sample}.bam</code></td>
<td><code>aligned/merged/{group}.bam</code> (samtools merge)</td>
</tr>
<tr>
<td>Scale factor</td>
<td>individual sample factor</td>
<td>mean of all samples in group</td>
</tr>
<tr>
<td>Factor source</td>
<td><code>resources/{group}_scaling_factors.tsv</code></td>
<td>same file, averaged</td>
</tr>
</tbody>
</table>
<h3 id="peak-calls-peaks">Peak Calls (<code>peaks/</code>)<a class="headerlink" href="#peak-calls-peaks" title="Anchor link to this section">&para;</a></h3>
<p>Peak calling results are simplified to 3-column BED files (chr, start, end):</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-6-1"><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>peaks/
</span><span id="__span-6-2"><a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>├── {method}/                  # macs2, macs3, homer, lanceotron, or seacr
</span><span id="__span-6-3"><a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>│   ├── {sample}.bed           # Individual sample peaks
</span><span id="__span-6-4"><a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a>│   └── merged/                # Consensus peaks across replicates
</span><span id="__span-6-5"><a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a>│       └── {group}.bed
</span></code></pre></div>
<p><strong>Supported Peak Callers:</strong></p>
<ul>
<li><strong>MACS2</strong>: Standard peak calling</li>
<li><strong>MACS3</strong>: Updated MACS peak calling</li>
<li><strong>HOMER</strong>: <code>findPeaks</code>-based peak calling</li>
<li><strong>LanceOtron</strong>: ML-based peak calling</li>
<li><strong>SEACR</strong>: Sparse Enrichment Analysis for CUT&amp;RUN/CUT&amp;Tag</li>
</ul>
<p><strong>Peak File Format:</strong></p>
<ul>
<li><strong>BED</strong>: Simple 3-column genomic coordinates (chr, start, end) extracted from each caller's native output</li>
</ul>
<h3 id="read-counts-readcounts">Read Counts (<code>readcounts/</code>)<a class="headerlink" href="#read-counts-readcounts" title="Anchor link to this section">&para;</a></h3>
<p>Quantification files for RNA-seq and CRISPR assays:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-7-1"><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>readcounts/
</span><span id="__span-7-2"><a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>├── feature_counts/
</span><span id="__span-7-3"><a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>│   └── read_counts.tsv         # Combined featureCounts output
</span><span id="__span-7-4"><a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>└── salmon/                     # RNA-seq only
</span><span id="__span-7-5"><a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>    └── salmon_counts.csv       # Combined Salmon quantification
</span></code></pre></div>
<h3 id="quality-control-qc">Quality Control (<code>qc/</code>)<a class="headerlink" href="#quality-control-qc" title="Anchor link to this section">&para;</a></h3>
<p>Comprehensive QC metrics:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-8-1"><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>qc/
</span><span id="__span-8-2"><a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>├── fastqc_raw/                     # FastQC reports on raw reads
</span><span id="__span-8-3"><a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a>│   ├── {sample}_1_fastqc.html      # Read 1 (paired-end)
</span><span id="__span-8-4"><a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a>│   └── {sample}_2_fastqc.html      # Read 2 (paired-end)
</span><span id="__span-8-5"><a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a>├── fastq_screen/                   # Contamination screening (if enabled)
</span><span id="__span-8-6"><a id="__codelineno-8-6" name="__codelineno-8-6" href="#__codelineno-8-6"></a>│   ├── {sample}_1_screen.html
</span><span id="__span-8-7"><a id="__codelineno-8-7" name="__codelineno-8-7" href="#__codelineno-8-7"></a>│   └── {sample}_2_screen.html
</span><span id="__span-8-8"><a id="__codelineno-8-8" name="__codelineno-8-8" href="#__codelineno-8-8"></a>├── qualimap_bamqc/                 # BAM quality metrics (non-RNA assays)
</span><span id="__span-8-9"><a id="__codelineno-8-9" name="__codelineno-8-9" href="#__codelineno-8-9"></a>│   └── {sample}/
</span><span id="__span-8-10"><a id="__codelineno-8-10" name="__codelineno-8-10" href="#__codelineno-8-10"></a>│       └── qualimapReport.html
</span><span id="__span-8-11"><a id="__codelineno-8-11" name="__codelineno-8-11" href="#__codelineno-8-11"></a>├── qualimap_rnaseq/                # RNA-specific BAM QC (RNA assay only)
</span><span id="__span-8-12"><a id="__codelineno-8-12" name="__codelineno-8-12" href="#__codelineno-8-12"></a>│   └── {sample}/
</span><span id="__span-8-13"><a id="__codelineno-8-13" name="__codelineno-8-13" href="#__codelineno-8-13"></a>│       └── qualimapReport.html
</span><span id="__span-8-14"><a id="__codelineno-8-14" name="__codelineno-8-14" href="#__codelineno-8-14"></a>├── alignment_stats.tsv             # Aggregated alignment statistics
</span><span id="__span-8-15"><a id="__codelineno-8-15" name="__codelineno-8-15" href="#__codelineno-8-15"></a>├── library_complexity/             # Picard duplicate metrics
</span><span id="__span-8-16"><a id="__codelineno-8-16" name="__codelineno-8-16" href="#__codelineno-8-16"></a>│   └── {sample}.metrics
</span><span id="__span-8-17"><a id="__codelineno-8-17" name="__codelineno-8-17" href="#__codelineno-8-17"></a>└── frip_enrichment/                # FRiP scores (if enabled, peak-calling assays)
</span><span id="__span-8-18"><a id="__codelineno-8-18" name="__codelineno-8-18" href="#__codelineno-8-18"></a>    └── {method}/
</span><span id="__span-8-19"><a id="__codelineno-8-19" name="__codelineno-8-19" href="#__codelineno-8-19"></a>        ├── {sample}_frip.pdf
</span><span id="__span-8-20"><a id="__codelineno-8-20" name="__codelineno-8-20" href="#__codelineno-8-20"></a>        └── {sample}_frip.txt
</span></code></pre></div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>For single-end data, FastQC and FastQ Screen files use <code>{sample}_fastqc.html</code> / <code>{sample}_screen.html</code> without the <code>_1</code>/<code>_2</code> suffix.</p>
</div>
<p><strong>QC Metrics Include:</strong></p>
<ul>
<li>Read quality scores per base position</li>
<li>GC content distribution</li>
<li>Adapter content</li>
<li>Duplication rates</li>
<li>Mapping statistics</li>
<li>Coverage uniformity</li>
</ul>
<h3 id="ucsc-genome-browser-hub-hub">UCSC Genome Browser Hub (<code>hub/</code>)<a class="headerlink" href="#ucsc-genome-browser-hub-hub" title="Anchor link to this section">&para;</a></h3>
<p>Ready-to-load UCSC track hub generated by TracKNado:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-9-1"><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>hub/
</span><span id="__span-9-2"><a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a>└── {hub_name}.hub.txt     # Hub description file (default: seqnado_hub.hub.txt)
</span></code></pre></div>
<p>The hub structure (genomes.txt, trackDb.txt, etc.) is generated by TracKNado and includes links to your bigWig and other track files.</p>
<p><strong>Usage:</strong></p>
<ol>
<li>Upload the <code>hub/</code> directory to a web-accessible location</li>
<li>Load in UCSC Genome Browser using the hub URL</li>
<li>Or use locally with IGV/other genome browsers</li>
</ol>
<h3 id="heatmaps-heatmap">Heatmaps (<code>heatmap/</code>)<a class="headerlink" href="#heatmaps-heatmap" title="Anchor link to this section">&para;</a></h3>
<p>DeepTools-generated heatmaps and metaplots (for assays with peak calling). One set of plots is produced per pileup method × scaling method combination, mirroring the bigwig structure:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-10-1"><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a>heatmap/
</span><span id="__span-10-2"><a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a>├── {method}/                        # deeptools, homer, or bamnado
</span><span id="__span-10-3"><a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a>│   ├── unscaled/
</span><span id="__span-10-4"><a id="__codelineno-10-4" name="__codelineno-10-4" href="#__codelineno-10-4"></a>│   │   ├── heatmap.pdf              # Signal heatmap over regions of interest
</span><span id="__span-10-5"><a id="__codelineno-10-5" name="__codelineno-10-5" href="#__codelineno-10-5"></a>│   │   └── metaplot.pdf             # Average signal profile
</span><span id="__span-10-6"><a id="__codelineno-10-6" name="__codelineno-10-6" href="#__codelineno-10-6"></a>│   ├── csaw/                        # If CSAW normalisation enabled
</span><span id="__span-10-7"><a id="__codelineno-10-7" name="__codelineno-10-7" href="#__codelineno-10-7"></a>│   │   ├── heatmap.pdf
</span><span id="__span-10-8"><a id="__codelineno-10-8" name="__codelineno-10-8" href="#__codelineno-10-8"></a>│   │   └── metaplot.pdf
</span><span id="__span-10-9"><a id="__codelineno-10-9" name="__codelineno-10-9" href="#__codelineno-10-9"></a>│   └── spikein/{spikein_method}/    # If spike-in normalisation enabled
</span><span id="__span-10-10"><a id="__codelineno-10-10" name="__codelineno-10-10" href="#__codelineno-10-10"></a>│       ├── heatmap.pdf
</span><span id="__span-10-11"><a id="__codelineno-10-11" name="__codelineno-10-11" href="#__codelineno-10-11"></a>│       └── metaplot.pdf
</span><span id="__span-10-12"><a id="__codelineno-10-12" name="__codelineno-10-12" href="#__codelineno-10-12"></a>└── merged/                          # If consensus groups defined
</span><span id="__span-10-13"><a id="__codelineno-10-13" name="__codelineno-10-13" href="#__codelineno-10-13"></a>    └── {method}/
</span><span id="__span-10-14"><a id="__codelineno-10-14" name="__codelineno-10-14" href="#__codelineno-10-14"></a>        ├── unscaled/
</span><span id="__span-10-15"><a id="__codelineno-10-15" name="__codelineno-10-15" href="#__codelineno-10-15"></a>        │   ├── heatmap.pdf
</span><span id="__span-10-16"><a id="__codelineno-10-16" name="__codelineno-10-16" href="#__codelineno-10-16"></a>        │   └── metaplot.pdf
</span><span id="__span-10-17"><a id="__codelineno-10-17" name="__codelineno-10-17" href="#__codelineno-10-17"></a>        └── csaw/                    # If CSAW normalisation enabled
</span><span id="__span-10-18"><a id="__codelineno-10-18" name="__codelineno-10-18" href="#__codelineno-10-18"></a>            ├── heatmap.pdf
</span><span id="__span-10-19"><a id="__codelineno-10-19" name="__codelineno-10-19" href="#__codelineno-10-19"></a>            └── metaplot.pdf
</span></code></pre></div>
<h3 id="motif-analysis-motifs">Motif Analysis (<code>motifs/</code>)<a class="headerlink" href="#motif-analysis-motifs" title="Anchor link to this section">&para;</a></h3>
<p>Motif enrichment analysis for peak-calling assays (if enabled):</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-11-1"><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>motifs/
</span><span id="__span-11-2"><a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a>├── homer/                      # HOMER findMotifsGenome results
</span><span id="__span-11-3"><a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a>│   └── {peak_method}/
</span><span id="__span-11-4"><a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a>│       └── {sample}/
</span><span id="__span-11-5"><a id="__codelineno-11-5" name="__codelineno-11-5" href="#__codelineno-11-5"></a>└── meme/                       # MEME-ChIP results (if enabled)
</span><span id="__span-11-6"><a id="__codelineno-11-6" name="__codelineno-11-6" href="#__codelineno-11-6"></a>    └── {peak_method}/
</span><span id="__span-11-7"><a id="__codelineno-11-7" name="__codelineno-11-7" href="#__codelineno-11-7"></a>        └── {sample}/
</span></code></pre></div>
<h3 id="genome-browser-plots-genome_browser_plots">Genome Browser Plots (<code>genome_browser_plots/</code>)<a class="headerlink" href="#genome-browser-plots-genome_browser_plots" title="Anchor link to this section">&para;</a></h3>
<p>Publication-ready visualisations generated with PlotNado (if configured with plotting coordinates). One set of plots is produced per pileup method × scaling method combination:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-12-1"><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a>genome_browser_plots/
</span><span id="__span-12-2"><a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a>├── {method}/                              # deeptools, homer, or bamnado
</span><span id="__span-12-3"><a id="__codelineno-12-3" name="__codelineno-12-3" href="#__codelineno-12-3"></a>│   ├── unscaled/
</span><span id="__span-12-4"><a id="__codelineno-12-4" name="__codelineno-12-4" href="#__codelineno-12-4"></a>│   │   ├── {region_name}.{format}         # Named regions from BED file
</span><span id="__span-12-5"><a id="__codelineno-12-5" name="__codelineno-12-5" href="#__codelineno-12-5"></a>│   │   ├── {chr}-{start}-{end}.{format}   # Unnamed regions use coordinates
</span><span id="__span-12-6"><a id="__codelineno-12-6" name="__codelineno-12-6" href="#__codelineno-12-6"></a>│   │   └── template.toml                  # PlotNado configuration template
</span><span id="__span-12-7"><a id="__codelineno-12-7" name="__codelineno-12-7" href="#__codelineno-12-7"></a>│   ├── csaw/                              # If CSAW normalisation enabled
</span><span id="__span-12-8"><a id="__codelineno-12-8" name="__codelineno-12-8" href="#__codelineno-12-8"></a>│   │   ├── {region_name}.{format}
</span><span id="__span-12-9"><a id="__codelineno-12-9" name="__codelineno-12-9" href="#__codelineno-12-9"></a>│   │   └── template.toml
</span><span id="__span-12-10"><a id="__codelineno-12-10" name="__codelineno-12-10" href="#__codelineno-12-10"></a>│   └── spikein/{spikein_method}/          # If spike-in normalisation enabled
</span><span id="__span-12-11"><a id="__codelineno-12-11" name="__codelineno-12-11" href="#__codelineno-12-11"></a>│       ├── {region_name}.{format}
</span><span id="__span-12-12"><a id="__codelineno-12-12" name="__codelineno-12-12" href="#__codelineno-12-12"></a>│       └── template.toml
</span><span id="__span-12-13"><a id="__codelineno-12-13" name="__codelineno-12-13" href="#__codelineno-12-13"></a>└── merged/                                # If consensus groups defined
</span><span id="__span-12-14"><a id="__codelineno-12-14" name="__codelineno-12-14" href="#__codelineno-12-14"></a>    └── {method}/
</span><span id="__span-12-15"><a id="__codelineno-12-15" name="__codelineno-12-15" href="#__codelineno-12-15"></a>        ├── unscaled/
</span><span id="__span-12-16"><a id="__codelineno-12-16" name="__codelineno-12-16" href="#__codelineno-12-16"></a>        │   ├── {region_name}.{format}
</span><span id="__span-12-17"><a id="__codelineno-12-17" name="__codelineno-12-17" href="#__codelineno-12-17"></a>        │   └── template.toml
</span><span id="__span-12-18"><a id="__codelineno-12-18" name="__codelineno-12-18" href="#__codelineno-12-18"></a>        └── csaw/                          # If CSAW normalisation enabled
</span><span id="__span-12-19"><a id="__codelineno-12-19" name="__codelineno-12-19" href="#__codelineno-12-19"></a>            ├── {region_name}.{format}
</span><span id="__span-12-20"><a id="__codelineno-12-20" name="__codelineno-12-20" href="#__codelineno-12-20"></a>            └── template.toml
</span></code></pre></div>
<p>Output format can be <code>svg</code>, <code>png</code>, or <code>pdf</code> as configured.</p>
<h3 id="geo-submission-geo_submission">GEO Submission (<code>geo_submission/</code>)<a class="headerlink" href="#geo-submission-geo_submission" title="Anchor link to this section">&para;</a></h3>
<p>Pre-formatted files for GEO/SRA submission (if enabled):</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-13-1"><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a>geo_submission/
</span><span id="__span-13-2"><a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a>├── samples_table.txt                 # Sample metadata (TSV format)
</span><span id="__span-13-3"><a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a>├── md5sums.txt                       # Combined checksums
</span><span id="__span-13-4"><a id="__codelineno-13-4" name="__codelineno-13-4" href="#__codelineno-13-4"></a>├── raw_data_checksums.txt            # Checksums for raw FASTQ files
</span><span id="__span-13-5"><a id="__codelineno-13-5" name="__codelineno-13-5" href="#__codelineno-13-5"></a>├── processed_data_checksums.txt      # Checksums for processed files
</span><span id="__span-13-6"><a id="__codelineno-13-6" name="__codelineno-13-6" href="#__codelineno-13-6"></a>├── upload_instructions.txt           # Instructions for GEO upload
</span><span id="__span-13-7"><a id="__codelineno-13-7" name="__codelineno-13-7" href="#__codelineno-13-7"></a>├── {sample}_1.fastq.gz               # Symlinks to raw FASTQ files
</span><span id="__span-13-8"><a id="__codelineno-13-8" name="__codelineno-13-8" href="#__codelineno-13-8"></a>├── {sample}_2.fastq.gz
</span><span id="__span-13-9"><a id="__codelineno-13-9" name="__codelineno-13-9" href="#__codelineno-13-9"></a>├── {sample}_{method}_{scale}.bigWig  # Renamed processed bigWig files
</span><span id="__span-13-10"><a id="__codelineno-13-10" name="__codelineno-13-10" href="#__codelineno-13-10"></a>├── {sample}_{method}.bed             # Renamed peak files
</span><span id="__span-13-11"><a id="__codelineno-13-11" name="__codelineno-13-11" href="#__codelineno-13-11"></a>└── {assay}/                          # Upload directory
</span></code></pre></div>
<h2 id="assay-specific-outputs">Assay-Specific Outputs<a class="headerlink" href="#assay-specific-outputs" title="Anchor link to this section">&para;</a></h2>
<h3 id="atac-seq">ATAC-seq<a class="headerlink" href="#atac-seq" title="Anchor link to this section">&para;</a></h3>
<p>ATAC-seq includes Tn5 insertion site correction during BAM processing and supports all peak callers. LanceOtron is the default peak caller.</p>
<p><strong>Key outputs:</strong></p>
<ul>
<li><code>aligned/{sample}.bam</code> -- Tn5-shifted, filtered alignments</li>
<li><code>peaks/lanceotron/{sample}.bed</code> -- ML-based peak calls (default)</li>
<li><code>bigwigs/{method}/{sample}.bigWig</code> -- Coverage tracks</li>
</ul>
<p><strong>Key Metrics:</strong></p>
<ul>
<li>Fragment size distribution (nucleosome periodicity visible in QC report)</li>
<li>FRiP (Fraction of Reads in Peaks) score (if enabled)</li>
</ul>
<h3 id="chip-seq">ChIP-seq<a class="headerlink" href="#chip-seq" title="Anchor link to this section">&para;</a></h3>
<p>Standard ChIP-seq with support for input controls:</p>
<p><strong>Key outputs:</strong></p>
<ul>
<li><code>aligned/{sample}.bam</code> -- Final alignments</li>
<li><code>peaks/{method}/{sample}.bed</code> -- Peak calls (MACS2, HOMER, LanceOtron)</li>
<li><code>peaks/{method}/merged/{group}.bed</code> -- Consensus peaks</li>
<li><code>tag_dirs/{sample}/</code> -- HOMER tag directories</li>
<li><code>motifs/homer/{method}/{sample}/</code> -- Motif analysis (if enabled)</li>
</ul>
<p><strong>Normalisation resources</strong> (if applicable):</p>
<ul>
<li><code>resources/{spikein_method}/normalisation_factors.json</code> -- Spike-in scaling factors</li>
<li><code>resources/binned_counts/read_counts.tsv</code> -- Genomic bin counts for CSAW</li>
<li><code>resources/{group}_scaling_factors.tsv</code> -- CSAW scaling factors per consensus group</li>
<li><code>bigwigs/deeptools/spikein/{spikein_method}/{sample}.bigWig</code> -- Spike-in normalised tracks</li>
<li><code>bigwigs/deeptools/csaw/{sample}.bigWig</code> -- CSAW normalised tracks</li>
</ul>
<h3 id="cuttag">CUT&amp;Tag<a class="headerlink" href="#cuttag" title="Anchor link to this section">&para;</a></h3>
<p>CUT&amp;Tag is a separate assay from ChIP-seq, with SEACR as the default peak caller and optional Tn5 shift correction:</p>
<p><strong>Key outputs:</strong></p>
<ul>
<li><code>aligned/{sample}.bam</code> -- Final alignments (optionally Tn5-shifted)</li>
<li><code>peaks/seacr/{sample}.bed</code> -- SEACR peak calls (default)</li>
<li><code>bigwigs/{method}/{sample}.bigWig</code> -- Coverage tracks</li>
</ul>
<h3 id="rna-seq">RNA-seq<a class="headerlink" href="#rna-seq" title="Anchor link to this section">&para;</a></h3>
<p>RNA-seq alignment uses STAR, with quantification by featureCounts and/or Salmon:</p>
<p><strong>Key outputs:</strong></p>
<ul>
<li><code>aligned/{sample}.bam</code> -- STAR-aligned, processed BAM</li>
<li><code>readcounts/feature_counts/read_counts.tsv</code> -- Combined gene-level counts</li>
<li><code>readcounts/salmon/salmon_counts.csv</code> -- Salmon quantification (if enabled)</li>
<li><code>bigwigs/{method}/{sample}_plus.bigWig</code> -- Stranded coverage (plus strand)</li>
<li><code>bigwigs/{method}/{sample}_minus.bigWig</code> -- Stranded coverage (minus strand)</li>
<li><code>qc/qualimap_rnaseq/{sample}/qualimapReport.html</code> -- RNA-specific QC</li>
</ul>
<h3 id="methylation-meth">Methylation (METH)<a class="headerlink" href="#methylation-meth" title="Anchor link to this section">&para;</a></h3>
<p>Methylation calling uses MethylDackel, with support for both bisulfite and TAPS methods:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-14-1"><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a>methylation/
</span><span id="__span-14-2"><a id="__codelineno-14-2" name="__codelineno-14-2" href="#__codelineno-14-2"></a>├── methyldackel/
</span><span id="__span-14-3"><a id="__codelineno-14-3" name="__codelineno-14-3" href="#__codelineno-14-3"></a>│   ├── {sample}_{genome}_CpG.bedGraph             # CpG methylation calls
</span><span id="__span-14-4"><a id="__codelineno-14-4" name="__codelineno-14-4" href="#__codelineno-14-4"></a>│   ├── {sample}_{genome}_CpG_inverted.bedGraph    # TAPS-inverted calls (TAPS method only)
</span><span id="__span-14-5"><a id="__codelineno-14-5" name="__codelineno-14-5" href="#__codelineno-14-5"></a>│   └── bias/
</span><span id="__span-14-6"><a id="__codelineno-14-6" name="__codelineno-14-6" href="#__codelineno-14-6"></a>│       └── {sample}_{genome}.txt                  # M-bias data
</span><span id="__span-14-7"><a id="__codelineno-14-7" name="__codelineno-14-7" href="#__codelineno-14-7"></a>├── methylation_conversion.tsv                     # Conversion rate statistics
</span><span id="__span-14-8"><a id="__codelineno-14-8" name="__codelineno-14-8" href="#__codelineno-14-8"></a>└── methylation_conversion.png                     # Conversion rate plot
</span></code></pre></div>
<p>Samples are split by genome (reference vs spike-in) with split BAMs at <code>aligned/spikein/{sample}_{genome}.bam</code>.</p>
<h3 id="snp-variant-calling">SNP / Variant Calling<a class="headerlink" href="#snp-variant-calling" title="Anchor link to this section">&para;</a></h3>
<p>Variant calling using bcftools:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-15-1"><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a>variant/
</span><span id="__span-15-2"><a id="__codelineno-15-2" name="__codelineno-15-2" href="#__codelineno-15-2"></a>├── {sample}.vcf.gz             # Called variants
</span><span id="__span-15-3"><a id="__codelineno-15-3" name="__codelineno-15-3" href="#__codelineno-15-3"></a>└── {sample}.anno.vcf.gz        # Annotated variants (if annotation enabled)
</span></code></pre></div>
<p>QC stats are produced at <code>qc/variant/{sample}.stats.txt</code>.</p>
<h3 id="micro-capture-c-mcc">Micro Capture-C (MCC)<a class="headerlink" href="#micro-capture-c-mcc" title="Anchor link to this section">&para;</a></h3>
<p>Chromatin contact analysis:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-16-1"><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a>mcc/
</span><span id="__span-16-2"><a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a>└── contacts/
</span><span id="__span-16-3"><a id="__codelineno-16-3" name="__codelineno-16-3" href="#__codelineno-16-3"></a>    └── {group}/
</span><span id="__span-16-4"><a id="__codelineno-16-4" name="__codelineno-16-4" href="#__codelineno-16-4"></a>        └── {group}.mcool      # Multi-resolution contact matrix
</span></code></pre></div>
<h3 id="crispr-screens">CRISPR Screens<a class="headerlink" href="#crispr-screens" title="Anchor link to this section">&para;</a></h3>
<p>Guide RNA quantification with optional MAGeCK analysis:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-17-1"><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a>readcounts/
</span><span id="__span-17-2"><a id="__codelineno-17-2" name="__codelineno-17-2" href="#__codelineno-17-2"></a>├── feature_counts/
</span><span id="__span-17-3"><a id="__codelineno-17-3" name="__codelineno-17-3" href="#__codelineno-17-3"></a>│   └── read_counts.tsv                     # featureCounts guide counts (always produced)
</span><span id="__span-17-4"><a id="__codelineno-17-4" name="__codelineno-17-4" href="#__codelineno-17-4"></a>└── mageck/                                 # MAGeCK analysis (if enabled)
</span><span id="__span-17-5"><a id="__codelineno-17-5" name="__codelineno-17-5" href="#__codelineno-17-5"></a>    ├── mageck_count.count.txt              # Raw guide counts
</span><span id="__span-17-6"><a id="__codelineno-17-6" name="__codelineno-17-6" href="#__codelineno-17-6"></a>    ├── mageck_count.count_normalized.txt   # Normalised counts
</span><span id="__span-17-7"><a id="__codelineno-17-7" name="__codelineno-17-7" href="#__codelineno-17-7"></a>    ├── mageck_count.countsummary.txt       # Count summary
</span><span id="__span-17-8"><a id="__codelineno-17-8" name="__codelineno-17-8" href="#__codelineno-17-8"></a>    ├── mageck_mle.gene_summary.txt         # MAGeCK MLE gene-level results
</span><span id="__span-17-9"><a id="__codelineno-17-9" name="__codelineno-17-9" href="#__codelineno-17-9"></a>    ├── mageck_mle.sgrna_summary.txt        # MAGeCK MLE sgRNA-level results
</span><span id="__span-17-10"><a id="__codelineno-17-10" name="__codelineno-17-10" href="#__codelineno-17-10"></a>    └── design_matrix.txt                   # Design matrix used
</span></code></pre></div>
<h2 id="accessing-your-results">Accessing Your Results<a class="headerlink" href="#accessing-your-results" title="Anchor link to this section">&para;</a></h2>
<h3 id="command-line">Command Line<a class="headerlink" href="#command-line" title="Anchor link to this section">&para;</a></h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-18-1"><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a><span class="c1"># Navigate to output directory</span>
</span><span id="__span-18-2"><a id="__codelineno-18-2" name="__codelineno-18-2" href="#__codelineno-18-2"></a><span class="nb">cd</span><span class="w"> </span>seqnado_output/
</span><span id="__span-18-3"><a id="__codelineno-18-3" name="__codelineno-18-3" href="#__codelineno-18-3"></a>
</span><span id="__span-18-4"><a id="__codelineno-18-4" name="__codelineno-18-4" href="#__codelineno-18-4"></a><span class="c1"># View main report</span>
</span><span id="__span-18-5"><a id="__codelineno-18-5" name="__codelineno-18-5" href="#__codelineno-18-5"></a>firefox<span class="w"> </span><span class="o">{</span>assay<span class="o">}</span>/seqnado_report.html<span class="w"> </span><span class="p">&amp;</span>
</span><span id="__span-18-6"><a id="__codelineno-18-6" name="__codelineno-18-6" href="#__codelineno-18-6"></a>
</span><span id="__span-18-7"><a id="__codelineno-18-7" name="__codelineno-18-7" href="#__codelineno-18-7"></a><span class="c1"># List peaks</span>
</span><span id="__span-18-8"><a id="__codelineno-18-8" name="__codelineno-18-8" href="#__codelineno-18-8"></a>ls<span class="w"> </span>-lh<span class="w"> </span><span class="o">{</span>assay<span class="o">}</span>/peaks/macs2/
</span><span id="__span-18-9"><a id="__codelineno-18-9" name="__codelineno-18-9" href="#__codelineno-18-9"></a>
</span><span id="__span-18-10"><a id="__codelineno-18-10" name="__codelineno-18-10" href="#__codelineno-18-10"></a><span class="c1"># Load BAM in IGV</span>
</span><span id="__span-18-11"><a id="__codelineno-18-11" name="__codelineno-18-11" href="#__codelineno-18-11"></a>igv<span class="w"> </span><span class="o">{</span>assay<span class="o">}</span>/aligned/<span class="o">{</span>sample<span class="o">}</span>.bam
</span></code></pre></div>
<h3 id="opening-reports">Opening Reports<a class="headerlink" href="#opening-reports" title="Anchor link to this section">&para;</a></h3>
<p>The HTML reports can be opened directly in your browser:</p>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-19-1"><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a><span class="c1"># On local machine</span>
</span><span id="__span-19-2"><a id="__codelineno-19-2" name="__codelineno-19-2" href="#__codelineno-19-2"></a>open<span class="w"> </span>seqnado_output/chip/seqnado_report.html
</span><span id="__span-19-3"><a id="__codelineno-19-3" name="__codelineno-19-3" href="#__codelineno-19-3"></a>
</span><span id="__span-19-4"><a id="__codelineno-19-4" name="__codelineno-19-4" href="#__codelineno-19-4"></a><span class="c1"># Via X11 forwarding on HPC</span>
</span><span id="__span-19-5"><a id="__codelineno-19-5" name="__codelineno-19-5" href="#__codelineno-19-5"></a>firefox<span class="w"> </span>seqnado_output/chip/seqnado_report.html<span class="w"> </span><span class="p">&amp;</span>
</span><span id="__span-19-6"><a id="__codelineno-19-6" name="__codelineno-19-6" href="#__codelineno-19-6"></a>
</span><span id="__span-19-7"><a id="__codelineno-19-7" name="__codelineno-19-7" href="#__codelineno-19-7"></a><span class="c1"># Transfer to local machine</span>
</span><span id="__span-19-8"><a id="__codelineno-19-8" name="__codelineno-19-8" href="#__codelineno-19-8"></a>scp<span class="w"> </span>-r<span class="w"> </span>user@hpc:path/to/seqnado_output/<span class="w"> </span>./
</span></code></pre></div>
<h2 id="finding-specific-outputs">Finding Specific Outputs<a class="headerlink" href="#finding-specific-outputs" title="Anchor link to this section">&para;</a></h2>
<h3 id="peak-calling-results">Peak calling results<a class="headerlink" href="#peak-calling-results" title="Anchor link to this section">&para;</a></h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-20-1"><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a>find<span class="w"> </span>seqnado_output/<span class="w"> </span>-name<span class="w"> </span><span class="s2">&quot;*.bed&quot;</span>
</span></code></pre></div>
<h3 id="coverage-tracks-for-visualisation">Coverage tracks for visualisation<a class="headerlink" href="#coverage-tracks-for-visualisation" title="Anchor link to this section">&para;</a></h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-21-1"><a id="__codelineno-21-1" name="__codelineno-21-1" href="#__codelineno-21-1"></a>find<span class="w"> </span>seqnado_output/<span class="w"> </span>-name<span class="w"> </span><span class="s2">&quot;*.bigWig&quot;</span>
</span></code></pre></div>
<h3 id="qc-html-reports">QC HTML reports<a class="headerlink" href="#qc-html-reports" title="Anchor link to this section">&para;</a></h3>
<div class="language-bash highlight"><pre><span></span><code><span id="__span-22-1"><a id="__codelineno-22-1" name="__codelineno-22-1" href="#__codelineno-22-1"></a>find<span class="w"> </span>seqnado_output/<span class="w"> </span>-name<span class="w"> </span><span class="s2">&quot;*.html&quot;</span>
</span></code></pre></div>
<hr />
<p><strong>See Also:</strong></p>
<ul>
<li><a href="../pipeline/">Pipeline Overview</a> - How outputs are generated</li>
<li><a href="../tools/">Tools Reference</a> - Understanding tool-specific outputs</li>
</ul>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../pipeline/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Pipeline Overview">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Pipeline Overview
              </div>
            </div>
          </a>
        
        
          
          <a href="../examples/" class="md-footer__link md-footer__link--next" aria-label="Next: Output Examples">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                Output Examples
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2024 - 2026 Milne Group
    </div>
  
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
    <a href="https://github.com/Milne-Group/SeqNado" target="_blank" rel="noopener" title="SeqNado on GitHub" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"/></svg>
    </a>
  
    
    
    
    
    <a href="https://pypi.org/project/seqnado" target="_blank" rel="noopener" title="SeqNado on PyPI" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.8 200.5c-7.7-30.9-22.3-54.2-53.4-54.2h-40.1v47.4c0 36.8-31.2 67.8-66.8 67.8H172.7c-29.2 0-53.4 25-53.4 54.3v101.8c0 29 25.2 46 53.4 54.3 33.8 9.9 66.3 11.7 106.8 0 26.9-7.8 53.4-23.5 53.4-54.3v-40.7H226.2v-13.6h160.2c31.1 0 42.6-21.7 53.4-54.2 11.2-33.5 10.7-65.7 0-108.6M286.2 444.7a20.4 20.4 0 1 1 0-40.7 20.4 20.4 0 1 1 0 40.7M167.8 248.1h106.8c29.7 0 53.4-24.5 53.4-54.3V91.9c0-29-24.4-50.7-53.4-55.6-35.8-5.9-74.7-5.6-106.8.1-45.2 8-53.4 24.7-53.4 55.6v40.7h106.9v13.6h-147c-31.1 0-58.3 18.7-66.8 54.2-9.8 40.7-10.2 66.1 0 108.6 7.6 31.6 25.7 54.2 56.8 54.2H101v-48.8c0-35.3 30.5-66.4 66.8-66.4m-6.6-183.4a20.4 20.4 0 1 1 0 40.8 20.4 20.4 0 1 1 0-40.8"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
      <div class="md-progress" data-md-component="progress" role="progressbar"></div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "..", "features": ["announce.dismiss", "content.action.edit", "content.action.view", "content.code.annotate", "content.code.copy", "content.tabs.link", "content.tooltips", "header.autohide", "navigation.expand", "navigation.footer", "navigation.indexes", "navigation.instant", "navigation.instant.prefetch", "navigation.instant.progress", "navigation.path", "navigation.sections", "navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "toc.follow"], "search": "../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script>
    
    
      <script src="../assets/javascripts/bundle.79ae519e.min.js"></script>
      
        <script src="../javascripts/toc-collapse.js"></script>
      
        <script src="../javascripts/mathjax.js"></script>
      
        <script src="https://unpkg.com/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>